<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Rachael Barron, Peggy Martinez, Marcelo Serpe, Sven Buerki" />


<title>Development of an in vitro method of propagation using growth regulators for Basin Big sagebrush (Artemisia tridentata subsp. tridentata) to support genome sequencing and GxE research</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.16/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sagebrush propagation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="protocols.html">Protocols</a>
</li>
<li>
  <a href="Analyses_rooting.html">Reproducible workflow</a>
</li>
<li>
  <a href="Gallery.html">Gallery</a>
</li>
<li>
  <a href="location_pop.html">Map</a>
</li>
<li>
  <a href="citation.html">How to cite</a>
</li>
<li>
  <a href="funding.html">Acknowledgements</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Development of an <em>in vitro</em> method of propagation using growth regulators for Basin Big sagebrush (<em>Artemisia tridentata</em> subsp. <em>tridentata</em>) to support genome sequencing and GxE research</h1>
<h4 class="author">Rachael Barron, Peggy Martinez, Marcelo Serpe, Sven Buerki</h4>

</div>


<div id="reproducible-workflow" class="section level1">
<h1><span class="header-section-number">1</span> Reproducible workflow</h1>
<div id="integrating-code-data-methods-and-key-outcomes" class="section level2">
<h2><span class="header-section-number">1.1</span> Integrating code, data, methods and key outcomes</h2>
<p>This document is associated to Barron et al. and provides the reproducible workflow integrating data, code, methods and results associated to this study.</p>
<p>Citations of <code>R</code> packages required to conduct this research and produce this document are provided in <a href="#appendix-1">Appendix 1</a>. Version information about <code>R</code>, the operating system (OS) and attached or <code>R</code> loaded packages are available in <a href="#appendix-2">Appendix 2</a>. The data underpinning this study are deposited on <a href="https://github.com/svenbuerki/Sagebrush_rooting_in_vitro_prop">GitHub</a>. Appendices together with the data and R code presented in this document should allow reproducibility of this study.</p>
</div>
</div>
<div id="overview-of-methodology" class="section level1">
<h1><span class="header-section-number">2</span> Overview of methodology</h1>
<p>In this document, we are presenting the code associated to the analyses performed on tissue culture data (rooting and calli) generated from shoot tips of diploid <em>Artemisia tridentata</em> subsp. <em>tridentata</em> (2n=2x=18).</p>
<p>The analyses were conducted in four steps:</p>
<ul>
<li><strong>Step 1:</strong> Present, import and tidy data (see section <a href="#step1">3</a>).</li>
<li><strong>Step 2:</strong> Preliminary analyses on rooting experiment (see section <a href="#step2">4</a>).</li>
<li><strong>Step 3:</strong> Comparative analyses on rooting data (see section <a href="#step3">5</a>). This was done following a three-tier approach:
<ul>
<li>Prepare response variables.</li>
<li>Statistical analyses (<em>p</em>-value &lt;= 0.01).</li>
<li>Clustering analyses.</li>
</ul></li>
<li><strong>Step 4:</strong> <em>In vitro</em> survival and growth of plantlets (see section <a href="#step4">6</a>).
<ul>
<li>Assessing the effect of growth regulators and rooting clusters on plantlet survival and growth rates.</li>
</ul></li>
</ul>
</div>
<div id="step1" class="section level1">
<h1><span class="header-section-number">3</span> Step 1: Present, import and tidy data</h1>
<div id="data-presentation" class="section level2">
<h2><span class="header-section-number">3.1</span> Data presentation</h2>
<p>The raw data used for this study are available in the <a href="https://github.com/svenbuerki/Sagebrush_rooting_in_vitro_prop/tree/master/01_Raw_Data">Sagebrush_rooting_in_vitro_prop</a> GitHub repository in the <code>01_Raw_Data</code> folder. The data can be split into two categories supporting analyses presented in this study:</p>
<ol style="list-style-type: decimal">
<li>Tissue culture data: These data are related to the rooting experiment conducted on shoot tips.</li>
<li><em>In vitro</em> survival and growth of plantlets: These data are related to the survival and growth experiment of rooted shoot tips into growth media.</li>
</ol>
<p>We are providing more information on these data below.</p>
<div id="codingind" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Coding of individual lines</h3>
<p>Figure <a href="#fig:mother">3.1</a> allows to better grasp the coding approach applied to identify individual lines in this study. Our coding protocol is as follows:</p>
<ul>
<li>Genotype (<code>G1</code>: drought-tolerant or <code>G2</code>: drought-sensitive). <code>G1</code> was collected in Idaho, whereas <code>G2</code> came from Utah (see <a href="https://svenbuerki.github.io/Sagebrush_rooting_in_vitro_prop/location_pop.html">Map</a> for more details.).</li>
<li>Magenta box ID (<code>b</code>). Multiple seeds from the same mother plant/individual were sown per Magenta box.</li>
<li>Individual in box. Because more than one plants were grown per magenta box, we have provided a unique ID for each plant.</li>
</ul>
<p>For instance, <code>G2_b27_1</code> corresponds to an individual line representing the drought-sensitive genotype (<code>G2</code>) grown in the Magenta box #27 (<code>b27</code>) and it was the first sampled plant in this box (<code>1</code>).</p>
<p>Overall, the combination of these three items allows providing a unique ID to each individual line included in this study.</p>
<div class="figure" style="text-align: center"><span id="fig:mother"></span>
<img src="03_Figures_report/MotherPlant_low_res.jpg" alt="Example of a Magenta box containing individuals of diploid sagebrush used for the rooting experiment." width="50%" />
<p class="caption">
Figure 3.1: Example of a Magenta box containing individuals of diploid sagebrush used for the rooting experiment.
</p>
</div>
</div>
<div id="tissue-culture-data" class="section level3 tabset tabset-pills">
<h3><span class="header-section-number">3.1.2</span> Tissue culture data</h3>
<div id="import-and-tidy-data" class="section level4">
<h4><span class="header-section-number">3.1.2.1</span> Import and tidy data</h4>
<p>The objective of this section is to merge data on rooting and callus stored in spreadsheets (in <code>csv</code> format). Each spreadsheet corresponds to a block and the code provided in section <a href="#tidy">3.1.2.2</a> collates these data into one <code>data.frame</code> for downstream analyses.</p>
<pre><code>## [1] &quot;List of csv files with block data:&quot;</code></pre>
<pre><code>## [1] &quot;01_Raw_Data/1_block_8_12_2020 - 1_block.csv&quot;
## [2] &quot;01_Raw_Data/2_block_8_15_2020 - 2_block.csv&quot;
## [3] &quot;01_Raw_Data/3_block_8_15_2020 - 3_block.csv&quot;
## [4] &quot;01_Raw_Data/4_block_8_16_2020 - 4_block.csv&quot;
## [5] &quot;01_Raw_Data/5_block_8_19_2020 - 5_block.csv&quot;</code></pre>
<pre><code>## [1] &quot;Merge files and print head of data.frame:&quot;</code></pre>
<pre><code>##   Block Treatment Replicate Genotype Individual Callus Root  IndGeno
## 1     1    IBA_05         1       G2      b20_1      0    0 G2_b20_1
## 2     1    IBA_05         2       G2      b20_1      1    3 G2_b20_1
## 3     1    IBA_05         3       G2      b20_1      1    0 G2_b20_1
## 4     1     IBA_1         1       G2      b20_1      1    2 G2_b20_1
## 5     1     IBA_1         2       G2      b20_1      1    4 G2_b20_1
## 6     1     IBA_1         3       G2      b20_1      1    1 G2_b20_1</code></pre>
<p>The final dataset (<code>MERGE</code>) contains the following columns (see Table <a href="#tab:statssampling">4.1</a> for more details on sampling):</p>
<ul>
<li><code>Block</code>: 1 to 5. Each individual were randomly allocated to a block and a block has 9 individual.</li>
<li><code>Treatment</code>: <code>Control</code> (= no growth regulator), <code>IBA_1</code> (IBA 1 mg/l), <code>IBA_05</code> (IBA 0.5 mg/l), <code>NAA_1</code> (NAA 1 mg/l), <code>NAA_05</code> (NAA 0.5 mg/l).</li>
<li><code>Replicate</code>: 1 to 3. Each individual line had 3 shoot tips randomly allocated per treatment.</li>
<li><code>Genotype</code>: <code>G1</code> (drought-tolerant genotype from ID3 population) and <code>G2</code> (drought-sensitive population from UT2 population). See distribution <a href="https://svenbuerki.github.io/Sagebrush_rooting_in_vitro_prop/location_pop.html">map</a>.</li>
<li>Individual line: <code>Individual</code> and <code>IndGeno</code>. The latter variable combines data on genotype and individual and it will be used throughout the document to represent the individual variable (see section <a href="#codingind">3.1.1</a> for more details).</li>
<li><code>Callus</code>: Binary (1 = present; 0 = absent).</li>
<li><code>Root</code>: Count of number of roots per shoot tip.</li>
</ul>
<p>The <code>MERGE</code> dataset is available in <a href="#appendix-3">Appendix 3</a> and can be directly there. Individual and merged processed files are available in the <code>02_Processed_data</code> folder (see file <code>Processed_blocks_rooting.csv</code> for merged data or click <a href="https://github.com/svenbuerki/Sagebrush_rooting_in_vitro_prop/blob/master/02_Processed_data/Processed_blocks_rooting.csv">here</a>).</p>
</div>
<div id="tidy" class="section level4">
<h4><span class="header-section-number">3.1.2.2</span> R code</h4>
<p>Raw data in <code>csv</code> format are stored in the <code>01_Raw_data</code> folder. The processed/tidy data are saved in <code>csv</code> format in the <code>02_Processed_data</code> folder. The <code>MERGE</code> object (in class <code>data.frame</code>) will be used for downstream analyses.</p>
<pre class="r"><code>###~~
#List all csv files (raw data from blocks)
###~~~

#Raw data are stored in the 01_Raw_Data folder
csv &lt;- list.files(path = &quot;01_Raw_Data&quot;, pattern=&#39;block.csv&#39;, full.names = T)

#Print names of csv files
print(csv)

#Rm processed files
toRm &lt;- grep(&quot;Processed_&quot;, csv)
if(length(toRm) &gt; 0){
  csv &lt;- csv[-toRm]
}

###~~~
#Execute loop to process all files and merge them (see below)
###~~~
# Empty object that will contain all processed data
# Save processed data into 02_Processed_data folder
MERGE &lt;- NULL
for(i in 1:length(csv)){
  ###
  #Read in sv file
  mat &lt;- read.csv(csv[i])
  
  ###~~~
  #Create final matrix
  ###~~~
  
  #List of individuals
  indcsv &lt;- LETTERS[seq( from = 1, to = 9 )]
  
  #Empty matrix
  FINALmat &lt;- data.frame(matrix(ncol=7, nrow = nrow(mat)*length(indcsv)))
  colnames(FINALmat) &lt;- c(&quot;Block&quot;,&quot;Treatment&quot;, &quot;Replicate&quot;, &quot;Genotype&quot;, &quot;Individual&quot;, &quot;Callus&quot;, &quot;Root&quot;)
  
  ###~~~
  #Start populating FINALmat 
  ###~~~
  # Add Block, Treatment and Replicates
  #Get data for block, treatment (and replicates)
  blockTreatRep &lt;- rep(as.vector(mat$X), 9)
  FINALmat$Block &lt;- sapply(strsplit(blockTreatRep, split=&#39;_&#39;), &quot;[&quot;, 1)
  FINALmat$Treatment &lt;- paste(sapply(strsplit(blockTreatRep, split=&#39;_&#39;), &quot;[&quot;, 2), sapply(strsplit(blockTreatRep, split=&#39;_&#39;), &quot;[&quot;, 3), sep=&#39;_&#39;)
  FINALmat$Replicate &lt;- sapply(strsplit(blockTreatRep, split=&#39;_&#39;), &quot;[&quot;, 4)
  
  #Deal with Controls
  contr &lt;- grep(&quot;Control_&quot;, FINALmat$Treatment)
  FINALmat$Replicate[contr] &lt;- sapply(strsplit(FINALmat$Treatment[contr], split=&quot;_&quot;), &quot;[&quot;,2)
  FINALmat$Treatment[contr] &lt;- sapply(strsplit(FINALmat$Treatment[contr], split=&quot;_&quot;), &quot;[&quot;,1)
  
  ###~~~
  # Fetch individual data
  ###~~~
  
  #Where are ind in mat?
  indcol &lt;- match(indcsv, colnames(mat))
  
  #Fetch raw data for each individual in block
  OUT &lt;- NULL
  for(j in 1:length(indcol)){
    #Extract info for each individual
    tmp &lt;- mat[,indcol[j]:(indcol[j]+2)]
    colnames(tmp) &lt;- c(&quot;Ind&quot;, &quot;Callus&quot;, &quot;Root&quot;)
    OUT &lt;- rbind(OUT, tmp)
  }
  
  ###~~~
  #Add OUT to FINALmat
  ###~~~
  
  #Genotypes
  FINALmat$Genotype &lt;- sapply(strsplit(as.vector(OUT$Ind), split=&#39;_&#39;), &quot;[&quot;, 1)
  #Ind
  FINALmat$Individual &lt;- paste(sapply(strsplit(as.vector(OUT$Ind), split=&#39;_&#39;), &quot;[&quot;, 2), sapply(strsplit(as.vector(OUT$Ind), split=&#39;_&#39;), &quot;[&quot;, 3), sep=&#39;_&#39;)
  #Callus
  FINALmat$Callus &lt;- OUT$Callus
  #Root
  FINALmat$Root &lt;- OUT$Root
  
  ###~~~
  #Save FINALmat
  ###~~~
  write.csv(FINALmat, file=paste0(&quot;02_Processed_data/Processed_&quot;, strsplit(csv[i], split=&quot;/&quot;)[[1]][2]), row.names = F, quote = F)
  
  ###~~~
  #MERGE all csv
  ###~~~
  MERGE &lt;- rbind(MERGE, FINALmat)
}

###~~~
#Finalize preping of the data
###~~~
# Add a binary rooting col
MERGE &lt;- data.frame(MERGE)

MERGE$IndGeno &lt;- paste(MERGE$Genotype, MERGE$Individual, sep=&quot;_&quot;)

#Create a binary for Root
MERGE$RootBin &lt;- MERGE$Root
MERGE$RootBin[MERGE$RootBin &gt; 0] &lt;- 1

###~~~
#Save MERGED files with all blocks
###~~~
# Save processed data into 02_Processed_data folder
write.csv(MERGE, file=paste0(&quot;02_Processed_data/Processed_&quot;, &quot;blocks_rooting.csv&quot;), row.names = F, quote = F)

#Return head of MERGE (see Appendix for more details)
head(MERGE)</code></pre>
</div>
</div>
<div id="in-vitro-survival-and-growth-of-plantlets" class="section level3 tabset tabset-pills">
<h3><span class="header-section-number">3.1.3</span> <em>In vitro</em> survival and growth of plantlets</h3>
<p>Data on survival and plantlets heights at 3 and 5 weeks are stored in the <code>01_Raw_Data</code> folder in <code>Survival_height_clones.csv</code>. The dataset is available in <a href="#appendix-4">Appendix 4</a> and can be directly downloaded there. Please find below the description of the variables:</p>
<ul>
<li><code>SeedID</code>: This is equal to <code>IndGeno</code> in <code>MERGE</code> and therefore refers to the rooted shoot tip that was transferred.</li>
<li><code>Cluster</code>: Rooting cluster as inferred by the clustering analysis.</li>
<li><code>X3_w_survival</code> and <code>X5_w_survival</code>: Binary survival (1 = alive, 0 = dead) data at 3 and 5 weeks.</li>
<li><code>X3_w_height</code> and <code>X5_w_height</code>: Continuous plantlets heights data (in cm) at 3 and 5 weeks.</li>
</ul>
</div>
</div>
</div>
<div id="step2" class="section level1">
<h1><span class="header-section-number">4</span> Step 2: Preliminary analyses on rooting experiment</h1>
<p>Preliminary data summarizing results of our experiments are provided here to assess the effect of growth regulators on <em>in vitro</em> rooting of <em>Artemisia tridentata</em>. During the course of the experiment, calli developed on a large proportion of shoot tips, which was unexpected based on preliminary data. We are therefore summarizing these data here by treatment and will devote a portion of the analyses to investigate whether callus development was associated to treatments and whether those inhibited or promoted rooting.</p>
<div id="statistics-on-plant-materials" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">4.1</span> Statistics on plant materials</h2>
<div id="analyses" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Analyses</h3>
<p>Table <a href="#tab:statssampling">4.1</a> provides a summary of plant materials included in this study. Overall, 45 individuals distributed into two genotypes (which were evenly sampled) were generated from seeds. For each individual line, 15 shoot tips were produced and included into five treatments with three replicates (= 3 shoot tips/ind./treatment). Individuals were randomly allocated to 5 blocks containing 9 individual each.</p>
<p>Overall, the sampling at the basis of the rooting experiment could summarized as follows:</p>
<blockquote>
<p>“45 individual lines x 5 treatments x 3 replicates, for n = 675 total shoot tips.”</p>
</blockquote>
<p>The code below generates Table <a href="#tab:statssampling">4.1</a>. The R code associated to producing Table <a href="#tab:statssampling">4.1</a> is available in section <a href="#statsampl">4.1.2</a>.</p>
<table>
<caption><span id="tab:statssampling">Table 4.1: </span>Summary of sampling at the basis of sagebrush rooting experiment.</caption>
<colgroup>
<col width="1%" />
<col width="2%" />
<col width="6%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="3%" />
<col width="39%" />
<col width="38%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">N. block</th>
<th align="left">N. treatments</th>
<th align="left">Treatments</th>
<th align="left">N. replicates</th>
<th align="left">N. genotypes</th>
<th align="left">N. individuals</th>
<th align="left">N. ind. per genotype</th>
<th align="left">Ind. G1</th>
<th align="left">Ind. G2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">5</td>
<td align="left">5</td>
<td align="left">Control, IBA_05, IBA_1, NAA_05, NAA_1</td>
<td align="left">3</td>
<td align="left">2</td>
<td align="left">45</td>
<td align="left">G1: 23, G2: 22</td>
<td align="left">G1_22_1, G1_b1_1, G1_b1_2, G1_b10_1, G1_b12_1, G1_b13_1, G1_b15_1, G1_b16_1, G1_b18_1, G1_b19_1, G1_b2_1, G1_b20_1, G1_b21_1, G1_b25_1, G1_b26_1, G1_b29_1, G1_b3_1, G1_b4_1, G1_b5_1, G1_b6_1, G1_b7_1, G1_b8_1, G1_b9_1</td>
<td align="left">G2_18_1, G2_29_1, G2_b11_1, G2_b15_1, G2_b16_1, G2_b17_1, G2_b19_1, G2_b2_1, G2_b20_1, G2_b21_1, G2_b21_2, G2_b24_1, G2_b25_1, G2_b26_1, G2_b27_1, G2_b27_2, G2_b28_1, G2_b4_1, G2_b5_1, G2_b6_1, G2_b7_1, G2_b8_1</td>
</tr>
</tbody>
</table>
</div>
<div id="statsampl" class="section level3">
<h3><span class="header-section-number">4.1.2</span> R code</h3>
<pre class="r"><code>#Create and populate table to summarize sampling (Table 1)
samp_mat &lt;- matrix(ncol=9, nrow=1)
colnames(samp_mat) &lt;- c(&quot;N. block&quot;, &quot;N. treatments&quot;, &quot;Treatments&quot;, &quot;N. replicates&quot;, &quot;N. genotypes&quot;, 
                        &quot;N. individuals&quot;, &quot;N. ind. per genotype&quot;, &quot;Ind. G1&quot;, &quot;Ind. G2&quot;)

#N block
samp_mat[,1] &lt;- length(unique(MERGE$Block))
#N treat
samp_mat[,2] &lt;- length(unique(MERGE$Treatment))
#Treat ID
samp_mat[,3] &lt;- paste(sort(unique(MERGE$Treatment)), collapse = &quot;, &quot;)
#N replicates
samp_mat[,4] &lt;- length(unique(MERGE$Replicate))
#N genotypes
samp_mat[,5] &lt;- length(unique(MERGE$Genotype))
#N ind
samp_mat[,6] &lt;- length(unique(MERGE$IndGeno))
#N ind per genotype
samp_mat[,7] &lt;- paste(paste(&quot;G1:&quot;, list(table(MERGE$Genotype)/15)[[1]][1]), paste(&quot;G2:&quot;, list(table(MERGE$Genotype)/15)[[1]][2]), sep=&#39;, &#39;)
#List ind G1
samp_mat[,8] &lt;- paste(sort(unique(subset(MERGE$IndGeno, MERGE$Genotype == &quot;G1&quot;))), collapse=&#39;, &#39;)
#List ind G2
samp_mat[,9] &lt;- paste(sort(unique(subset(MERGE$IndGeno, MERGE$Genotype == &quot;G2&quot;))), collapse=&#39;, &#39;)

#Write table
#write.csv(samp_mat, file=&quot;Table_1_sampling_summary.csv&quot;, row.names = F)

#Plot table
knitr::kable(samp_mat, caption = &quot;Summary of sampling at the basis of sagebrush rooting experiment.&quot;)</code></pre>
</div>
</div>
<div id="effect-of-growth-regulators-on-in-vitro-callus-development-in-artemisia-tridentata" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">4.2</span> Effect of growth regulators on <em>in vitro</em> callus development in <em>Artemisia tridentata</em></h2>
<div id="analyses-1" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Analyses</h3>
<p>The effect of growth regulators on callus development is provided in Table <a href="#tab:growthbyregulatorcallus">4.2</a> and compared to the control treatment (= no growth regulator). These data show contrasting calli development between the control and treatments including growth hormones. Control shoot tips had limited calli development, whereas responses for those undergoing treatments were very high (&gt;70%; Table <a href="#tab:growthbyregulatorcallus">4.2</a>). The R code associated to these analyses is presented in section <a href="#growthbyregulatorcallus">4.2.2</a>.</p>
<table>
<caption><span id="tab:growthbyregulatorcallus">Table 4.2: </span>Effect of growth regulators on in vitro calli development on shoot tips of diploid Artemisia tridentata subsp. tridentata. Response: mean of the three replicates.</caption>
<thead>
<tr class="header">
<th align="left">Growth regulator</th>
<th align="left">Concentration (mg/l)</th>
<th align="right">Response (%)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">2.96</td>
</tr>
<tr class="even">
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">75.56</td>
</tr>
<tr class="odd">
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">84.44</td>
</tr>
<tr class="even">
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">81.48</td>
</tr>
<tr class="odd">
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">87.41</td>
</tr>
</tbody>
</table>
</div>
<div id="growthbyregulatorcallus" class="section level3">
<h3><span class="header-section-number">4.2.2</span> R code</h3>
<pre class="r"><code>#Matrix summarizing effect of growth regulators on rooting
CallusStatbyTreat &lt;- matrix(ncol=3, nrow=5)
colnames(CallusStatbyTreat) &lt;- c(&quot;Growth regulator&quot;, &quot;Concentration (mg/l)&quot;, &quot;Response (%)&quot;)
rownames(CallusStatbyTreat) &lt;- sort(unique(MERGE$Treatment))

#Convert to dataframe
CallusStatbyTreat &lt;- as.data.frame(CallusStatbyTreat)

#Populate matrix
CallusStatbyTreat$`Growth regulator` &lt;- sapply(strsplit(rownames(CallusStatbyTreat), split=&quot;_&quot;),&quot;[[&quot;,1)
CallusStatbyTreat$`Concentration (mg/l)` &lt;- c(&quot;-&quot;, sapply(strsplit(rownames(CallusStatbyTreat)[2:nrow(CallusStatbyTreat)], split=&quot;_&quot;),&quot;[[&quot;,2))

#Infer Response and mean no roots
for(i in 1:nrow(CallusStatbyTreat)){
  foo &lt;- subset(MERGE, MERGE$Treatment == rownames(CallusStatbyTreat)[i])
  CallusStatbyTreat$`Response (%)`[i] &lt;- round(100*mean(foo$Callus),2) 
}


#Plot table
knitr::kable(CallusStatbyTreat, row.names=F, caption = &quot;Effect of growth regulators on in vitro calli development on shoot tips of diploid Artemisia tridentata subsp. tridentata. Response: mean of the three replicates.&quot;)</code></pre>
</div>
</div>
<div id="effect-of-growth-regulators-on-in-vitro-rooting-of-artemisia-tridentata" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">4.3</span> Effect of growth regulators on <em>in vitro</em> rooting of <em>Artemisia tridentata</em></h2>
<div id="analyses-2" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Analyses</h3>
<p>The effect of growth regulators is provided in Table <a href="#tab:growthbyregulator">4.3</a> and compared to the control treatment (= no growth regulator). These data show a very high level of variation in rooting (see column <code>Av. No. roots</code>). We are hypothesizing that this trend is caused by an individual effect. To test this hypothesis we will conduct comparative analyses (see <strong>Step 4</strong>) to sort individuals based on their rooting abilities. Finally, the effect of growth regulators on <em>in vitro</em> rooting will be compared within each cluster. The R code associated to these analyses is presented in section <a href="#growthbyregulatorcode">4.3.2</a>.</p>
<table>
<caption><span id="tab:growthbyregulator">Table 4.3: </span>Effect of growth regulators on in vitro rooting for shoot tips of diploid Artemisia tridentata subsp. tridentata. Response: mean of the three replicates; Av. No. of roots (Average number of roots per shoot tip): means.</caption>
<thead>
<tr class="header">
<th align="left">Growth regulator</th>
<th align="left">Concentration (mg/l)</th>
<th align="right">Response (%)</th>
<th align="right">Av. No. of roots</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">8.89</td>
<td align="right">0.18</td>
</tr>
<tr class="even">
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">47.41</td>
<td align="right">1.37</td>
</tr>
<tr class="odd">
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">60.00</td>
<td align="right">1.87</td>
</tr>
<tr class="even">
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">40.74</td>
<td align="right">0.79</td>
</tr>
<tr class="odd">
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">40.00</td>
<td align="right">1.06</td>
</tr>
</tbody>
</table>
</div>
<div id="growthbyregulatorcode" class="section level3">
<h3><span class="header-section-number">4.3.2</span> R code</h3>
<pre class="r"><code>#Matrix summarizing effect of growth regulators on rooting
RootingStatbyTreat &lt;- matrix(ncol=4, nrow=5)
colnames(RootingStatbyTreat) &lt;- c(&quot;Growth regulator&quot;, &quot;Concentration (mg/l)&quot;, &quot;Response (%)&quot;, &quot;Av. No. of roots&quot;)
rownames(RootingStatbyTreat) &lt;- sort(unique(MERGE$Treatment))

#Convert to dataframe
RootingStatbyTreat &lt;- as.data.frame(RootingStatbyTreat)

#Populate matrix
RootingStatbyTreat$`Growth regulator` &lt;- sapply(strsplit(rownames(RootingStatbyTreat), split=&quot;_&quot;),&quot;[[&quot;,1)
RootingStatbyTreat$`Concentration (mg/l)` &lt;- c(&quot;-&quot;, sapply(strsplit(rownames(RootingStatbyTreat)[2:nrow(RootingStatbyTreat)], split=&quot;_&quot;),&quot;[[&quot;,2))

#Infer Response and mean no roots
for(i in 1:nrow(RootingStatbyTreat)){
  foo &lt;- subset(MERGE, MERGE$Treatment == rownames(RootingStatbyTreat)[i])
  RootingStatbyTreat$`Response (%)`[i] &lt;- paste(round(100*(mean(foo$RootBin)),2), &quot;+/-&quot;, round(sd(foo$RootBin),2), sep=&quot; &quot;)
  RootingStatbyTreat$`Av. No. of roots`[i] &lt;- paste(round(mean(foo$Root),2), &quot;+/-&quot;, round(sd(foo$Root),2), sep=&quot; &quot;)
}


#Plot table
knitr::kable(RootingStatbyTreat, row.names=F, caption = &quot;Effect of growth regulators on in vitro rooting for shoot tips of diploid Artemisia tridentata subsp. tridentata. Response: mean of the three replicates +/- standard deviation; Av. No. of roots (Average number of roots per shoot tip): mean +/- standard deviation.&quot;)</code></pre>
</div>
</div>
</div>
<div id="step3" class="section level1">
<h1><span class="header-section-number">5</span> Step 3: Comparative analyses</h1>
<p>The input data for these analyses are stored in <code>MERGE</code>, but we will transform the data to test for treatment, genotype and individual effects.</p>
<div id="prepare-the-response-variables" class="section level2">
<h2><span class="header-section-number">5.1</span> Prepare the response variables</h2>
<div id="treatment" class="section level3 tabset tabset-pills">
<h3><span class="header-section-number">5.1.1</span> Treatment</h3>
<p>The effect of growth regulators (= treatment) will be assessed based on three response variables: the number of shoot tips per plate that formed callus, those that developed roots (presence/absence), and the total number of roots. The analyses and associated R code (see section <a href="#codeTreatdata">5.1.1.2</a>) to produce these variables based on <code>MERGE</code> is found below.</p>
<div id="analyses-3" class="section level4">
<h4><span class="header-section-number">5.1.1.1</span> Analyses</h4>
<p>We are displaying callus and rooting responses by plate unit to test for treatment effect. Here, a plate corresponds to an experimental unit, which is equal to a treatment by block by replicate (basically each row in the table is a plate). In a nutshell, the R code subset <code>MERGE</code> by plate and counts the number of callus (binary), root (binary) and root (count) (see section <a href="#codeTreatdata">5.1.1.2</a> for more details).</p>
<div id="htmlwidget-dda6897f5d1eb673e353" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-dda6897f5d1eb673e353">{"x":{"filter":"none","caption":"<caption>Data for treatment analyses sorted by plates.<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75"],["Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1"],["1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5"],["1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3"],[0,0,0,1,0,0,2,0,0,0,0,0,0,1,0,7,9,9,9,9,7,4,7,3,6,6,7,8,4,7,9,9,9,9,8,9,6,4,7,9,8,7,7,6,7,9,7,9,9,8,9,5,6,6,8,7,6,7,8,6,6,9,9,9,9,9,7,6,4,9,8,9,9,8,7],[1,2,1,0,0,3,0,2,1,0,1,1,0,0,0,2,6,4,5,6,6,3,5,4,2,3,3,7,4,4,7,6,6,4,7,4,4,4,6,5,6,6,4,6,6,3,3,7,8,6,7,4,4,2,2,3,1,2,0,3,5,3,4,2,6,7,3,1,3,2,4,5,4,3,2],[3,2,5,0,0,5,0,4,3,0,1,1,0,0,0,3,15,10,19,21,15,13,20,8,6,9,8,24,8,6,28,26,12,5,15,4,12,12,27,26,26,19,9,17,15,8,8,15,11,12,13,17,5,2,5,4,1,2,0,4,14,13,15,9,13,21,5,4,5,3,12,8,14,3,4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Treatment<\/th>\n      <th>Block<\/th>\n      <th>Replicate<\/th>\n      <th>Callus<\/th>\n      <th>RootBin<\/th>\n      <th>RootCount<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="codeTreatdata" class="section level4">
<h4><span class="header-section-number">5.1.1.2</span> R code</h4>
<div id="htmlwidget-97bcb68ab7571818cb21" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-97bcb68ab7571818cb21">{"x":{"filter":"none","caption":"<caption>Data for treatment analyses sorted by plates.<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75"],["Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1","NAA_1"],["1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","1","1","1","2","2","2","3","3","3","4","4","4","5","5","5"],["1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3"],[0,0,0,1,0,0,2,0,0,0,0,0,0,1,0,7,9,9,9,9,7,4,7,3,6,6,7,8,4,7,9,9,9,9,8,9,6,4,7,9,8,7,7,6,7,9,7,9,9,8,9,5,6,6,8,7,6,7,8,6,6,9,9,9,9,9,7,6,4,9,8,9,9,8,7],[1,2,1,0,0,3,0,2,1,0,1,1,0,0,0,2,6,4,5,6,6,3,5,4,2,3,3,7,4,4,7,6,6,4,7,4,4,4,6,5,6,6,4,6,6,3,3,7,8,6,7,4,4,2,2,3,1,2,0,3,5,3,4,2,6,7,3,1,3,2,4,5,4,3,2],[3,2,5,0,0,5,0,4,3,0,1,1,0,0,0,3,15,10,19,21,15,13,20,8,6,9,8,24,8,6,28,26,12,5,15,4,12,12,27,26,26,19,9,17,15,8,8,15,11,12,13,17,5,2,5,4,1,2,0,4,14,13,15,9,13,21,5,4,5,3,12,8,14,3,4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Treatment<\/th>\n      <th>Block<\/th>\n      <th>Replicate<\/th>\n      <th>Callus<\/th>\n      <th>RootBin<\/th>\n      <th>RootCount<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="genotype" class="section level3 tabset tabset-pills">
<h3><span class="header-section-number">5.1.2</span> Genotype</h3>
<p>The effect of genotypes will be assessed based on three response variables: the number of shoot tips per plate that formed callus, those that developed roots (presence/absence), and the total number of roots. The analyses and associated R code (see section <a href="#codeGenodata">5.1.2.2</a>) to produce these variables based on <code>MERGE</code> is found below.</p>
<div id="analyses-4" class="section level4">
<h4><span class="header-section-number">5.1.2.1</span> Analyses</h4>
<p>We are displaying callus and rooting responses by individual to test for treatment effect. See section <a href="#codeGenodata">5.1.2.2</a> for more details on R code.</p>
<div id="htmlwidget-cbdc8de69c20097394c4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cbdc8de69c20097394c4">{"x":{"filter":"none","caption":"<caption>Data for genotype analyses sorted by individuals.<\/caption>","data":[["2","3","8","10","12","16","17","19","21","22","23","24","25","26","27","32","38","39","40","42","43","44","45","1","4","5","6","7","9","11","13","14","15","18","20","28","29","30","31","33","34","35","36","37","41"],["G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2"],["G1_b19_1","G1_b16_1","G1_b6_1","G1_b8_1","G1_b21_1","G1_b4_1","G1_b29_1","G1_b26_1","G1_b5_1","G1_b15_1","G1_b3_1","G1_b13_1","G1_b12_1","G1_b10_1","G1_b7_1","G1_22_1","G1_b9_1","G1_b1_1","G1_b1_2","G1_b25_1","G1_b20_1","G1_b2_1","G1_b18_1","G2_b20_1","G2_b27_1","G2_b8_1","G2_b15_1","G2_b26_1","G2_b2_1","G2_b4_1","G2_b21_1","G2_b21_2","G2_b7_1","G2_b6_1","G2_18_1","G2_b17_1","G2_b28_1","G2_b19_1","G2_b27_2","G2_b16_1","G2_b11_1","G2_29_1","G2_b24_1","G2_b25_1","G2_b5_1"],[11,10,10,12,10,12,12,10,12,4,11,12,3,10,0,11,10,9,10,11,11,8,11,11,12,11,12,12,12,11,12,12,12,12,5,7,10,12,8,8,10,12,12,11,4],[7,10,6,11,7,5,9,2,5,1,12,12,0,9,0,0,6,3,7,7,6,9,4,8,13,8,3,2,3,10,10,6,11,2,5,5,2,1,6,5,8,6,11,1,2],[23,27,10,21,11,7,18,2,9,1,44,46,0,28,0,0,15,9,12,20,12,26,8,20,72,14,4,2,5,40,29,15,19,3,7,7,8,1,23,9,23,14,44,2,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Genotype<\/th>\n      <th>IndGeno<\/th>\n      <th>Callus<\/th>\n      <th>RootBin<\/th>\n      <th>RootCount<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="codeGenodata" class="section level4">
<h4><span class="header-section-number">5.1.2.2</span> R code</h4>
<pre class="r"><code>#Prepare an empty matrix to be populated 

#How many rows? This is equal to number of individual lines
NrowGen &lt;- length(unique(MERGE$IndGeno))

#Create empty matrix
TreatGeno &lt;- matrix(ncol=5, nrow = NrowGen)
colnames(TreatGeno) &lt;- c(&quot;Genotype&quot;, &quot;IndGeno&quot;, &quot;Callus&quot;,&quot;RootBin&quot;,&quot;RootCount&quot;)

#Populate with Genotype, Individual
TreatGeno[,2] &lt;- unique(MERGE$IndGeno)
TreatGeno[,1] &lt;- sapply(strsplit(unique(MERGE$IndGeno), split=&quot;_&quot;), &quot;[[&quot;, 1)

#Get Callus Bina, Root Bin and Root Cout
for(i in 1:nrow(TreatGeno)){
  tmp &lt;- subset(MERGE, MERGE$Genotype == TreatGeno[i,1] &amp; MERGE$IndGeno == TreatGeno[i,2])
  #Callus Binary
  TreatGeno[i,3] &lt;- sum(as.numeric(tmp$Callus))
  #Root Binary
  TreatGeno[i,4] &lt;- sum(as.numeric(tmp$RootBin))
  #Root Count
  TreatGeno[i,5] &lt;- sum(as.numeric(tmp$Root))
}
#Convert into data.frame
TreatGeno &lt;- as.data.frame(TreatGeno)

#Order by genotype
TreatGeno &lt;- TreatGeno[order(TreatGeno$Genotype),]

#Change class of response variables to numerical
i &lt;- c(3, 4, 5) 
TreatGeno[ , i] &lt;- apply(TreatGeno[ , i], 2, function(x) as.numeric(as.character(x)))

#Plot table
DT::datatable(TreatGeno, caption = &quot;Data for genotype analyses sorted by individuals.&quot;)</code></pre>
</div>
</div>
<div id="individual" class="section level3 tabset tabset-pills">
<h3><span class="header-section-number">5.1.3</span> Individual</h3>
<div id="analyses-5" class="section level4">
<h4><span class="header-section-number">5.1.3.1</span> Analyses</h4>
<p>Here, we are preparing contingency tables for callus (binary) and root (binary) data to test for individual effect. Analyses on root (count) data will be directly performed on the <code>MERGE</code> dataset. See section <a href="#codeInddata">5.1.3.2</a> for associated R code.</p>
<div id="htmlwidget-af70b4cb61154727c47f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-af70b4cb61154727c47f">{"x":{"filter":"none","caption":"<caption>Callus data for individual analyses (0 = absence; 1 = presence).<\/caption>","data":[["G1_22_1","G1_b1_1","G1_b1_2","G1_b10_1","G1_b12_1","G1_b13_1","G1_b15_1","G1_b16_1","G1_b18_1","G1_b19_1","G1_b2_1","G1_b20_1","G1_b21_1","G1_b25_1","G1_b26_1","G1_b29_1","G1_b3_1","G1_b4_1","G1_b5_1","G1_b6_1","G1_b7_1","G1_b8_1","G1_b9_1","G2_18_1","G2_29_1","G2_b11_1","G2_b15_1","G2_b16_1","G2_b17_1","G2_b19_1","G2_b2_1","G2_b20_1","G2_b21_1","G2_b21_2","G2_b24_1","G2_b25_1","G2_b26_1","G2_b27_1","G2_b27_2","G2_b28_1","G2_b4_1","G2_b5_1","G2_b6_1","G2_b7_1","G2_b8_1"],[4,6,5,5,12,3,11,5,4,4,7,4,5,4,5,3,4,3,3,5,15,3,5,10,3,5,3,7,8,3,3,4,3,3,3,4,3,3,7,5,4,11,3,3,4],[11,9,10,10,3,12,4,10,11,11,8,11,10,11,10,12,11,12,12,10,0,12,10,5,12,10,12,8,7,12,12,11,12,12,12,11,12,12,8,10,11,4,12,12,11]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>0<\/th>\n      <th>1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-87b9d411ff26d8419b71" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-87b9d411ff26d8419b71">{"x":{"filter":"none","caption":"<caption>Root data for individual analyses (0 = absence; 1 = presence).<\/caption>","data":[["G1_22_1","G1_b1_1","G1_b1_2","G1_b10_1","G1_b12_1","G1_b13_1","G1_b15_1","G1_b16_1","G1_b18_1","G1_b19_1","G1_b2_1","G1_b20_1","G1_b21_1","G1_b25_1","G1_b26_1","G1_b29_1","G1_b3_1","G1_b4_1","G1_b5_1","G1_b6_1","G1_b7_1","G1_b8_1","G1_b9_1","G2_18_1","G2_29_1","G2_b11_1","G2_b15_1","G2_b16_1","G2_b17_1","G2_b19_1","G2_b2_1","G2_b20_1","G2_b21_1","G2_b21_2","G2_b24_1","G2_b25_1","G2_b26_1","G2_b27_1","G2_b27_2","G2_b28_1","G2_b4_1","G2_b5_1","G2_b6_1","G2_b7_1","G2_b8_1"],[15,12,8,6,15,3,14,5,11,8,6,9,8,8,13,6,3,10,10,9,15,4,9,10,9,7,12,10,10,14,12,7,5,9,4,14,13,2,9,13,5,13,13,4,7],[0,3,7,9,0,12,1,10,4,7,9,6,7,7,2,9,12,5,5,6,0,11,6,5,6,8,3,5,5,1,3,8,10,6,11,1,2,13,6,2,10,2,2,11,8]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>0<\/th>\n      <th>1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="codeInddata" class="section level4">
<h4><span class="header-section-number">5.1.3.2</span> R code</h4>
<pre class="r"><code>###~~~
#Callus
###~~~
#Binary for each individual line
IndCallBin &lt;- table(MERGE$IndGeno, MERGE$Callus)

#Plot table
DT::datatable(as.data.frame.matrix(IndCallBin), caption = &quot;Callus data for individual analyses (0 = absence; 1 = presence).&quot;)

###~~~
#Root
###~~~
#Binary for each individual line
IndRootBin &lt;- table(MERGE$IndGeno, MERGE$RootBin)

#Plot table
DT::datatable(as.data.frame.matrix(IndRootBin), caption = &quot;Root data for individual analyses (0 = absence; 1 = presence).&quot;)</code></pre>
</div>
</div>
</div>
<div id="testing-for-treatment-effect" class="section level2">
<h2><span class="header-section-number">5.2</span> Testing for treatment effect</h2>
<p>The effect of growth regulators (= treatments) is assessed on three response variables stored in <code>TreatMat</code>: the number of shoot tips per plate that formed callus (<code>Callus</code>), those that developed roots (presence/absence; <code>RootBin</code>), and the total number of roots (<code>RootCount</code>). The latter served as a proxy for the number of roots per shoot tip. The presence/absence of callus and roots was evaluated using a generalized linear model (GLM) with Poisson distributed errors. The goodness of fit of the model was verified with a chi-square test. The treatments’ impact on the total number of roots was analyzed using a negative binomial generalized linear model (GLMNB). The deviance goodness of fit test ascertained models’ fitness, as judged by <em>p</em>-values above 0.05. Post-hoc multiple comparisons were analyzed with Tukey’s significant difference test. The GLM, GLMNB, and post-hoc analyses were conducted in R using the <em>glm</em>, <em>glm.nb</em>, and <em>emmeans</em> functions, respectively.</p>
<div id="callus" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Callus</h3>
<div id="glm-model" class="section level4">
<h4><span class="header-section-number">5.2.1.1</span> GLM model</h4>
<pre class="r"><code>###~~~
#GLM poisson model on Callus data
###~~~
m1 &lt;- glm(Callus ~ Treatment, family = poisson(link = &quot;log&quot;), data=TreatMat)
summary (m1)</code></pre>
<pre><code>## 
## Call:
## glm(formula = Callus ~ Treatment, family = poisson(link = &quot;log&quot;), 
##     data = TreatMat)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -1.6402  -0.6949   0.0474   0.3949   2.1431  
## 
## Coefficients:
##                 Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)      -1.3218     0.5000  -2.644  0.00821 ** 
## TreatmentIBA_05   3.2387     0.5097   6.354 2.10e-10 ***
## TreatmentIBA_1    3.3499     0.5087   6.585 4.54e-11 ***
## TreatmentNAA_05   3.3142     0.5090   6.511 7.46e-11 ***
## TreatmentNAA_1    3.3844     0.5084   6.657 2.80e-11 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 201.074  on 74  degrees of freedom
## Residual deviance:  34.487  on 70  degrees of freedom
## AIC: 281.2
## 
## Number of Fisher Scoring iterations: 6</code></pre>
<pre class="r"><code>###~~~
#Test for goodness of fit of deviance
###~~~
pchisq(m1$deviance, m1$df.residual, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.9998869</code></pre>
</div>
<div id="post-hoc-tukeys-significant-difference-test" class="section level4">
<h4><span class="header-section-number">5.2.1.2</span> Post-hoc: Tukey’s significant difference test</h4>
<p>The <em>emmeans</em> function implemented in the <code>emmeans</code> R package <span class="citation">(Lenth <a href="#ref-R-emmeans">2020</a>)</span> was used to compare treatments (with a <em>p</em>-value &lt;= 0.01):</p>
<pre class="r"><code># Estimated marginal means (Least-squares means) for treatments using m1
posthocCallus &lt;- emmeans(m1, specs = pairwise ~ Treatment, type = &quot;response&quot;)

# Identify significant pairwise comparisons
signCalTreat &lt;- as.data.frame(posthocCallus$contrasts)
signCalTreat &lt;- signCalTreat[which(signCalTreat$p.value &lt;= 0.01),c(1:3,5,6)]

#Print table
knitr::kable(signCalTreat, row.names=F, caption = &quot;Significant (p-value &lt;= 0.01) treatment pairwise comparisons based on callus binary data.&quot;)</code></pre>
<table>
<caption><span id="tab:emmeansCalBin">Table 5.1: </span>Significant (p-value &lt;= 0.01) treatment pairwise comparisons based on callus binary data.</caption>
<thead>
<tr class="header">
<th align="left">contrast</th>
<th align="right">ratio</th>
<th align="right">SE</th>
<th align="right">z.ratio</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Control / IBA_05</td>
<td align="right">0.0392157</td>
<td align="right">0.0199886</td>
<td align="right">-6.353968</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Control / IBA_1</td>
<td align="right">0.0350877</td>
<td align="right">0.0178490</td>
<td align="right">-6.585274</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Control / NAA_05</td>
<td align="right">0.0363636</td>
<td align="right">0.0185094</td>
<td align="right">-6.511047</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Control / NAA_1</td>
<td align="right">0.0338983</td>
<td align="right">0.0172340</td>
<td align="right">-6.656893</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Table <a href="#tab:emmeansCalBin">5.1</a> summarizes statistical comparisons of treatments focusing on significant effects. These analyses are not entirely conclusive, but they demonstrate that:</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>Control is significantly different than the other treatments (<code>***</code>). Here, very limited calli production was observed in shoot tips included in the control treatment.</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>No significant differences between IBA_1, IBA_05, NAA_01 and NAA_05.</li>
</ol></li>
</ul>
</div>
</div>
<div id="root" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Root</h3>
<p>Here, we will analyze treatment effect on rooting using the binary (<code>RootBin</code>) and count (<code>Root</code>) data.</p>
<div id="binary-data" class="section level4">
<h4><span class="header-section-number">5.2.2.1</span> Binary data</h4>
<div id="glm-model-1" class="section level5">
<h5><span class="header-section-number">5.2.2.1.1</span> GLM model</h5>
<pre class="r"><code>###~~~
#GLM poisson model on Root binary data
###~~~
m1root &lt;- glm(RootBin ~ TreatMat$Treatment, family = poisson(link = &quot;log&quot;), data=TreatMat)
summary (m1root)</code></pre>
<pre><code>## 
## Call:
## glm(formula = RootBin ~ TreatMat$Treatment, family = poisson(link = &quot;log&quot;), 
##     data = TreatMat)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -2.7080  -0.6481  -0.1305   0.2996   1.9534  
## 
## Coefficients:
##                          Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)               -0.2231     0.2887  -0.773     0.44    
## TreatMat$TreatmentIBA_05   1.6740     0.3146   5.321 1.03e-07 ***
## TreatMat$TreatmentIBA_1    1.9095     0.3093   6.173 6.68e-10 ***
## TreatMat$TreatmentNAA_05   1.5224     0.3186   4.778 1.77e-06 ***
## TreatMat$TreatmentNAA_1    1.5041     0.3191   4.713 2.44e-06 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 123.787  on 74  degrees of freedom
## Residual deviance:  62.494  on 70  degrees of freedom
## AIC: 282.33
## 
## Number of Fisher Scoring iterations: 5</code></pre>
<pre class="r"><code>###~~~
#Test for goodness of fit of deviance
###~~~
pchisq(m1root$deviance, m1root$df.residual, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.726308</code></pre>
</div>
<div id="post-hoc-tukeys-significant-difference-test-1" class="section level5">
<h5><span class="header-section-number">5.2.2.1.2</span> Post-hoc: Tukey’s significant difference test</h5>
<p>The <em>emmeans</em> function implemented in the <code>emmeans</code> R package <span class="citation">(Lenth <a href="#ref-R-emmeans">2020</a>)</span> was used to compare treatments (with a <em>p</em>-value &lt;= 0.01):</p>
<pre class="r"><code># Estimated marginal means (Least-squares means) for treatments using m1root
posthocRootBin &lt;- emmeans(m1root, specs = pairwise ~ Treatment, type = &quot;response&quot;)

# Identify significant pairwise comparisons
signRtTreat &lt;- as.data.frame(posthocRootBin$contrasts)
signRtTreat &lt;- signRtTreat[which(signRtTreat$p.value &lt;= 0.01),c(1:3,5,6)]

#Print table
knitr::kable(signRtTreat, row.names=F, caption = &quot;Significant (p-value &lt;= 0.01) treatment pairwise comparisons based on root binary data.&quot;)</code></pre>
<table>
<caption><span id="tab:emmeansRootBin">Table 5.2: </span>Significant (p-value &lt;= 0.01) treatment pairwise comparisons based on root binary data.</caption>
<thead>
<tr class="header">
<th align="left">contrast</th>
<th align="right">ratio</th>
<th align="right">SE</th>
<th align="right">z.ratio</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Control / IBA_05</td>
<td align="right">0.1875000</td>
<td align="right">0.0589826</td>
<td align="right">-5.321412</td>
<td align="right">1.00e-06</td>
</tr>
<tr class="even">
<td align="left">Control / IBA_1</td>
<td align="right">0.1481481</td>
<td align="right">0.0458248</td>
<td align="right">-6.173406</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="odd">
<td align="left">Control / NAA_05</td>
<td align="right">0.2181818</td>
<td align="right">0.0695153</td>
<td align="right">-4.778313</td>
<td align="right">1.75e-05</td>
</tr>
<tr class="even">
<td align="left">Control / NAA_1</td>
<td align="right">0.2222222</td>
<td align="right">0.0709199</td>
<td align="right">-4.712912</td>
<td align="right">2.41e-05</td>
</tr>
</tbody>
</table>
<p>Table <a href="#tab:emmeansRootBin">5.2</a> summarizes statistical comparison of treatments focusing on significant effects. These analyses are not entirely conclusive, but they demonstrate that:</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>Control is significantly different than the other treatments (<code>***</code>).</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>No significant differences between IBA_1, IBA_05, NAA_01 and NAA_05.</li>
</ol></li>
</ul>
</div>
</div>
<div id="count-data" class="section level4">
<h4><span class="header-section-number">5.2.2.2</span> Count data</h4>
<div id="glm-negative-binomial-model" class="section level5">
<h5><span class="header-section-number">5.2.2.2.1</span> GLM negative binomial model</h5>
<p>The <em>glm.nb</em> function is implemented in the <code>MASS</code> R package.</p>
<pre class="r"><code>###~~~
#GLM negative binomial poisson model on Root count data
###~~~
m2root &lt;- glm.nb(RootCount ~ Treatment, data= TreatMat)
summary (m2root)</code></pre>
<pre><code>## 
## Call:
## glm.nb(formula = RootCount ~ Treatment, data = TreatMat, init.theta = 3.6577786, 
##     link = log)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -2.8133  -1.0458  -0.2747   0.7926   1.6608  
## 
## Coefficients:
##                 Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)       0.4700     0.2447   1.921   0.0548 .  
## TreatmentIBA_05   2.0423     0.2890   7.067 1.59e-12 ***
## TreatmentIBA_1    2.3553     0.2865   8.222  &lt; 2e-16 ***
## TreatmentNAA_05   1.4948     0.2957   5.054 4.32e-07 ***
## TreatmentNAA_1    1.7848     0.2917   6.118 9.49e-10 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for Negative Binomial(3.6578) family taken to be 1)
## 
##     Null deviance: 171.310  on 74  degrees of freedom
## Residual deviance:  89.514  on 70  degrees of freedom
## AIC: 450.15
## 
## Number of Fisher Scoring iterations: 1
## 
## 
##               Theta:  3.658 
##           Std. Err.:  0.955 
## 
##  2 x log-likelihood:  -438.147</code></pre>
<pre class="r"><code>m3root &lt;- glm.nb(RootCount ~ Treatment + Block, data= TreatMat)
summary (m3root)</code></pre>
<pre><code>## 
## Call:
## glm.nb(formula = RootCount ~ Treatment + Block, data = TreatMat, 
##     init.theta = 4.396626204, link = log)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -2.5067  -0.9103  -0.1958   0.5221   2.2197  
## 
## Coefficients:
##                 Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)      0.71639    0.27406   2.614  0.00895 ** 
## TreatmentIBA_05  2.09891    0.28166   7.452 9.20e-14 ***
## TreatmentIBA_1   2.43701    0.27891   8.738  &lt; 2e-16 ***
## TreatmentNAA_05  1.47559    0.28975   5.093 3.53e-07 ***
## TreatmentNAA_1   1.79915    0.28496   6.314 2.73e-10 ***
## Block2          -0.05933    0.21334  -0.278  0.78095    
## Block3          -0.30275    0.21749  -1.392  0.16391    
## Block4          -0.48743    0.22122  -2.203  0.02757 *  
## Block5          -0.66866    0.22547  -2.966  0.00302 ** 
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for Negative Binomial(4.3966) family taken to be 1)
## 
##     Null deviance: 189.968  on 74  degrees of freedom
## Residual deviance:  85.974  on 66  degrees of freedom
## AIC: 446.19
## 
## Number of Fisher Scoring iterations: 1
## 
## 
##               Theta:  4.40 
##           Std. Err.:  1.17 
## 
##  2 x log-likelihood:  -426.191</code></pre>
<pre class="r"><code>###~~~
#Test for goodness of fit of deviance
###~~~
print(&quot;M3 better fits data compared to M2. M3 will be used for Tukey&#39;s tests.&quot;)</code></pre>
<pre><code>## [1] &quot;M3 better fits data compared to M2. M3 will be used for Tukey&#39;s tests.&quot;</code></pre>
<pre class="r"><code>#M2
pchisq(m2root$deviance, m2root$df.residual, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.05796607</code></pre>
<pre class="r"><code>#M3
pchisq(m3root$deviance, m2root$df.residual, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.09432734</code></pre>
</div>
<div id="post-hoc-tukeys-significant-difference-test-2" class="section level5">
<h5><span class="header-section-number">5.2.2.2.2</span> Post-hoc: Tukey’s significant difference test</h5>
<p>The <em>emmeans</em> function implemented in the <code>emmeans</code> R package <span class="citation">(Lenth <a href="#ref-R-emmeans">2020</a>)</span> was used to compare treatments (with a <em>p</em>-value &lt;= 0.01):</p>
<pre class="r"><code># Estimated marginal means (Least-squares means) for treatments using m2root
posthocRootCout &lt;- emmeans(m3root, specs = pairwise ~ Treatment, type = &quot;response&quot;)

# Identify significant pairwise comparisons
signRtCtTreat &lt;- as.data.frame(posthocRootCout$contrasts)
signRtCtTreat &lt;- signRtCtTreat[which(round(signRtCtTreat$p.value,3) &lt;= 0.015),c(1:3,5,6)]

#Print table
knitr::kable(signRtCtTreat, row.names=F, caption = &quot;Significant (p-value &lt;= 0.01) treatment pairwise comparisons based on root count data.&quot;)</code></pre>
<table>
<caption><span id="tab:emmeansRootcount">Table 5.3: </span>Significant (p-value &lt;= 0.01) treatment pairwise comparisons based on root count data.</caption>
<thead>
<tr class="header">
<th align="left">contrast</th>
<th align="right">ratio</th>
<th align="right">SE</th>
<th align="right">z.ratio</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Control / IBA_05</td>
<td align="right">0.1225900</td>
<td align="right">0.0345285</td>
<td align="right">-7.451969</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Control / IBA_1</td>
<td align="right">0.0874223</td>
<td align="right">0.0243829</td>
<td align="right">-8.737608</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">Control / NAA_05</td>
<td align="right">0.2286428</td>
<td align="right">0.0662503</td>
<td align="right">-5.092565</td>
<td align="right">0.0000035</td>
</tr>
<tr class="even">
<td align="left">Control / NAA_1</td>
<td align="right">0.1654402</td>
<td align="right">0.0471444</td>
<td align="right">-6.313607</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">IBA_1 / NAA_05</td>
<td align="right">2.6153833</td>
<td align="right">0.5512108</td>
<td align="right">4.561698</td>
<td align="right">0.0000498</td>
</tr>
<tr class="even">
<td align="left">IBA_1 / NAA_1</td>
<td align="right">1.8924267</td>
<td align="right">0.3861679</td>
<td align="right">3.125851</td>
<td align="right">0.0152653</td>
</tr>
</tbody>
</table>
<p>Table <a href="#tab:emmeansRootcount">5.3</a> summarizes statistical comparison of treatments focusing on significant effects. The ranking of treatments is as follows:</p>
<ul>
<li>(a): <code>IBA1</code></li>
<li>(ab): <code>IBA_05</code></li>
<li>(b): <code>NAA_1</code> and <code>NAA_05</code></li>
<li>(c): <code>Control</code></li>
</ul>
</div>
</div>
</div>
</div>
<div id="testing-for-genotype-effect" class="section level2">
<h2><span class="header-section-number">5.3</span> Testing for genotype effect</h2>
<p>The effect of genotypes is assessed on three response variables stored in <code>TreatGeno</code>: the number of shoot tips per plate that formed callus (<code>Callus</code>), those that developed roots (presence/absence; <code>RootBin</code>), and the total number of roots (<code>RootCount</code>). The latter served as a proxy for the number of roots per shoot tip. The presence/absence of callus and roots was evaluated using a generalized linear model (GLM) with Poisson distributed errors. The goodness of fit of the model was verified with a chi-square test. The treatments’ impact on the total number of roots was analyzed using a negative binomial generalized linear model (GLMNB). The deviance goodness of fit test ascertained models’ fitness, as judged by <em>p</em>-values above 0.05. Post-hoc multiple comparisons were analyzed with Tukey’s significant difference test. The GLM, GLMNB, and post-hoc analyses were conducted in R using the <em>glm</em>, <em>glm.nb</em>, and <em>emmeans</em> functions, respectively.</p>
<div id="callus-1" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Callus</h3>
<div id="glm-model-2" class="section level4">
<h4><span class="header-section-number">5.3.1.1</span> GLM model</h4>
<pre class="r"><code>###~~~
#GLM poisson model on Callus data
###~~~
m1CalGeno &lt;- glm(Callus ~ Genotype, family = poisson(link = &quot;log&quot;), data=TreatGeno)
summary (m1CalGeno)</code></pre>
<pre><code>## 
## Call:
## glm(formula = Callus ~ Genotype, family = poisson(link = &quot;log&quot;), 
##     data = TreatGeno)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -4.3738  -0.1136   0.1957   0.4957   0.7570  
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)  2.25813    0.06742  33.494   &lt;2e-16 ***
## GenotypeG2   0.08017    0.09451   0.848    0.396    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 48.554  on 44  degrees of freedom
## Residual deviance: 47.834  on 43  degrees of freedom
## AIC: 233.82
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<pre class="r"><code>###~~~
#Test for goodness of fit of deviance
###~~~
pchisq(m1CalGeno$deviance, m1CalGeno$df.residual, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.2828945</code></pre>
</div>
<div id="post-hoc-tukeys-significant-difference-test-3" class="section level4">
<h4><span class="header-section-number">5.3.1.2</span> Post-hoc: Tukey’s significant difference test</h4>
<p>The <em>emmeans</em> function implemented in the <code>emmeans</code> R package <span class="citation">(Lenth <a href="#ref-R-emmeans">2020</a>)</span> was used to compare treatments (with a <em>p</em>-value &lt;= 0.01):</p>
<pre class="r"><code># Estimated marginal means (Least-squares means) for treatments using m1CalGeno
posthocCallusGeno &lt;- emmeans(m1CalGeno, specs = pairwise ~ Genotype, type = &quot;response&quot;)

# Identify significant pairwise comparisons
signCalGeno &lt;- as.data.frame(posthocCallusGeno$contrasts)
signCalGeno &lt;- signCalGeno[which(signCalGeno$p.value &lt;= 0.01),c(1:3,5,6)]

#Print table
print(&quot;No significant difference between genotypes!&quot;)</code></pre>
<pre><code>## [1] &quot;No significant difference between genotypes!&quot;</code></pre>
</div>
</div>
<div id="root-1" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Root</h3>
<p>Here, we will analyze treatment effect on rooting using the binary (<code>RootBin</code>) and count (<code>Root</code>) data. Both variables will be analyzed with the negative binomial model implemented in <em>glm.nb</em> function.</p>
<div id="binary-data-1" class="section level4">
<h4><span class="header-section-number">5.3.2.1</span> Binary data</h4>
<div id="glm-negative-binomial-model-1" class="section level5">
<h5><span class="header-section-number">5.3.2.1.1</span> GLM negative binomial model</h5>
<pre class="r"><code>###~~~
#GLM negative binomial poisson model on Root binary data
###~~~
m1root &lt;- glm.nb(RootBin ~ Genotype, data= TreatGeno)
summary (m1root)</code></pre>
<pre><code>## 
## Call:
## glm.nb(formula = RootBin ~ Genotype, data = TreatGeno, init.theta = 3.428456238, 
##     link = log)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -2.6337  -0.8748   0.0000   0.6569   1.4207  
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)  1.79176    0.14117  12.693   &lt;2e-16 ***
## GenotypeG2  -0.03077    0.20248  -0.152    0.879    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for Negative Binomial(3.4285) family taken to be 1)
## 
##     Null deviance: 53.240  on 44  degrees of freedom
## Residual deviance: 53.217  on 43  degrees of freedom
## AIC: 249.81
## 
## Number of Fisher Scoring iterations: 1
## 
## 
##               Theta:  3.43 
##           Std. Err.:  1.28 
## 
##  2 x log-likelihood:  -243.806</code></pre>
<pre class="r"><code>#Print table
print(paste(&quot;No significant difference between genotypes!&quot;, &quot;No need to pursue analyses&quot;, sep=&quot;\n&quot;))</code></pre>
<pre><code>## [1] &quot;No significant difference between genotypes!\nNo need to pursue analyses&quot;</code></pre>
<pre class="r"><code>###~~~
#Test for goodness of fit of deviance
###~~~
pchisq(m1root$deviance, m1root$df.residual, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.1366434</code></pre>
</div>
</div>
<div id="count-data-1" class="section level4">
<h4><span class="header-section-number">5.3.2.2</span> Count data</h4>
<div id="glm-negative-binomial-model-2" class="section level5">
<h5><span class="header-section-number">5.3.2.2.1</span> GLM negative binomial model</h5>
<pre class="r"><code>###~~~
#GLM negative binomial poisson model on Root count data
###~~~
m2root &lt;- glm.nb(RootCount ~ Genotype, data= TreatGeno)
summary(m2root)</code></pre>
<pre><code>## 
## Call:
## glm.nb(formula = RootCount ~ Genotype, data = TreatGeno, init.theta = 1.063463775, 
##     link = log)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -2.4078  -0.9789  -0.2245   0.3525   1.9624  
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)  2.71958    0.20916   13.00   &lt;2e-16 ***
## GenotypeG2   0.08378    0.29874    0.28    0.779    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for Negative Binomial(1.0635) family taken to be 1)
## 
##     Null deviance: 52.243  on 44  degrees of freedom
## Residual deviance: 52.164  on 43  degrees of freedom
## AIC: 347.16
## 
## Number of Fisher Scoring iterations: 1
## 
## 
##               Theta:  1.063 
##           Std. Err.:  0.236 
## 
##  2 x log-likelihood:  -341.163</code></pre>
<pre class="r"><code>#Print table
print(paste(&quot;No significant difference between genotypes!&quot;, &quot;No need to pursue analyses&quot;, sep=&quot;\n&quot;))</code></pre>
<pre><code>## [1] &quot;No significant difference between genotypes!\nNo need to pursue analyses&quot;</code></pre>
<pre class="r"><code>###~~~
#Test for goodness of fit of deviance
###~~~
pchisq(m2root$deviance, m2root$df.residual, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.1595328</code></pre>
</div>
</div>
</div>
</div>
<div id="testing-for-individual-effect" class="section level2">
<h2><span class="header-section-number">5.4</span> Testing for individual effect</h2>
<p>Here, we will start by testing for individual effects by conducting chi-square analyses on binary callus and root data stored respectively in <code>IndCallBin</code> and <code>IndRootBin</code>. These analyses will be further investigated by conducting clustering analyses (see section <a href="#clustAnalyses">5.5.1</a>).</p>
<p>To identify top performers, we will then apply the Kruskal-Wallis rank sum test on root count data followed by pairwise Wilcoxon tests based on the <code>MERGE</code> dataset. We use a non-parametric test since the data are not normaly distributed:</p>
<pre class="r"><code># Shapiro-Wilk normality test on Root count data in MERGE
shapiro.test(MERGE$Root)</code></pre>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  MERGE$Root
## W = 0.62681, p-value &lt; 2.2e-16</code></pre>
<p>To qualify as a top performer, an individual will have to outperform &gt;= 20% of individuals (based on pairwise Wilcoxon tests with a <em>p</em>-value &lt;= 0.01).</p>
<div id="callus-and-root-binary-data" class="section level3">
<h3><span class="header-section-number">5.4.1</span> Callus and Root binary data</h3>
<pre class="r"><code>###~~~
#Callus binary data
###~~~
chisq.test(t(IndCallBin))</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  t(IndCallBin)
## X-squared = 102.12, df = 44, p-value = 1.61e-06</code></pre>
<pre class="r"><code>###~~~
#Root binary data
###~~~
chisq.test(t(IndRootBin))</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  t(IndRootBin)
## X-squared = 163.51, df = 44, p-value = 1.189e-15</code></pre>
<p>Both analyses strongly suggest individual effects, which will be further investigated in the clustering analyses (see section <a href="#clustAnalyses">5.5.1</a>).</p>
</div>
<div id="root-count-data" class="section level3 tabset tabset-pills">
<h3><span class="header-section-number">5.4.2</span> Root count data</h3>
<div id="analyses-6" class="section level4">
<h4><span class="header-section-number">5.4.2.1</span> Analyses</h4>
<p>Here, we are applying the Kruskal-Wallis rank sum test followed by pairwise Wilcoxon tests to assess individual effects and rank those (with a <em>p</em>-value of &lt;= 0.01) on <code>MERGE</code> data. Results of these analyses are presented below and the R code is available in section <a href="#RInd">5.4.2.2</a>. To qualify as a top performer, an individual has to outperform &gt;= 20% of individuals (based on pairwise Wilcoxon tests with a <em>p</em>-value &lt;= 0.01).</p>
<pre><code>## [1] &quot;Perform Kruskal-Wallis Rank Sum Test: Root ~ Individual&quot;</code></pre>
<pre><code>## 
##  Kruskal-Wallis rank sum test
## 
## data:  Root by IndGeno
## Kruskal-Wallis chi-squared = 184.14, df = 44, p-value &lt; 2.2e-16</code></pre>
<pre><code>## [1] &quot;Perform Wilcoxon test on Root ~ Individual&quot;</code></pre>
<table>
<caption><span id="tab:kruskalInd">Table 5.4: </span>List of individual that are outperforming at least 20% of individuals (based on Wilcoxon rank sum tests on Individual variable with a pval &lt;= 0.01).</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Percentage of individuals outperformed</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G2_b27_1</td>
<td align="right">48.9</td>
</tr>
<tr class="even">
<td align="left">G2_b7_1</td>
<td align="right">20.0</td>
</tr>
</tbody>
</table>
<p>The analyses reported in Table <a href="#tab:kruskalInd">5.4</a> demonstrate that only two individuals (G2_b27_1, G2_b7_1, which are referred to as P1 and P2) are significantly outperforming at least 20% of the individuals based on their rooting abilities independently to treatment.</p>
</div>
<div id="RInd" class="section level4">
<h4><span class="header-section-number">5.4.2.2</span> R code</h4>
<pre class="r"><code>###~~~
# Run Kruskal-Wallis Rank Sum Test
###~~~

print(&quot;Perform Kruskal-Wallis Rank Sum Test: Root ~ Individual&quot;)
K1 &lt;- kruskal.test(Root ~ IndGeno, data = MERGE)
print(K1)

###~~~
# Run Wilcoxon test on Root ~ IndGeno
###~~~

print(&quot;Perform Wilcoxon test on Root ~ Individual&quot;)
wilcoxRootInd &lt;- pairwise.wilcox.test(MERGE$Root, MERGE$IndGeno,
                 p.adjust.method = &quot;BH&quot;)

# Only select data.frame (distance matrix)
wilcoxRootInd &lt;- wilcoxRootInd$p.value

# Convert distance matrix into 3 cols: ind1, ind2, p-val
OUT &lt;- NULL
for(i in 1:nrow(wilcoxRootInd)){
  tmp &lt;- wilcoxRootInd[i,]
  tmp &lt;- cbind(rep(rownames(wilcoxRootInd)[i], length(tmp)), names(tmp), as.vector(tmp))
  OUT &lt;- rbind(OUT, tmp)
}
wilcoxSimp &lt;- as.data.frame(OUT)
colnames(wilcoxSimp) &lt;- c(&quot;Ind1&quot;, &quot;Ind2&quot;, &quot;P-val&quot;)

# Tidy data
wilcoxSimp &lt;- as.matrix(wilcoxSimp[-which(is.na(wilcoxSimp$`P-val`) == T),])

# Individual effect
IndRoot &lt;- as.data.frame(wilcoxSimp[which(as.numeric(wilcoxSimp[,3]) &lt;= 0.01),])

# Infer mean rooting per individual
meanRoot &lt;- aggregate(Root ~ IndGeno, mean, data = MERGE)

# Diff of means Ind1-Ind2
IndRoot$Diff_mean &lt;- meanRoot[match(IndRoot[,1], meanRoot[,1]),2]-meanRoot[match(IndRoot[,2], meanRoot[,1]),2]

# What are the top performers (= most efficient at producing roots)?
IndRootOut &lt;- IndRoot[which(IndRoot$Diff_mean &gt; 0),]

# Percentage of individuals that are outperformed by P1
TopInd &lt;- 100*(sort(table(IndRootOut[,1]), decreasing=T)/length(unique(MERGE$IndGeno)))

# Select only ind that are outperforming at least 20% of individuals 
TopInd &lt;- TopInd[which(TopInd &gt;= 20)]

TopInd &lt;- as.matrix(round(TopInd, 1))
colnames(TopInd) &lt;- c(&quot;Percentage of individuals outperformed&quot;)

# Plot table
knitr::kable(TopInd, caption = &quot;List of individual that are outperforming at least 20% of individuals (based on Wilcoxon rank sum tests on Individual variable with a pval &lt;= 0.01).&quot;)</code></pre>
</div>
</div>
</div>
<div id="clustering-analyses-on-rooting-data" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">5.5</span> Clustering analyses on rooting data</h2>
<p>In this section, we are applying a clustering approach on callus (binary) and root (count) data to further investigate individual effects independently to treatments. We first conduct independent clustering analyses on the two variables to sort individual lines based on their abilities to promote callus development and roots. These analyses will allow identifying callus and root clusters, which will then be compared by plotting those on the respective clustering analyses. To further characterize the clusters, ridgeline plots (or density plots) will be inferred for each variable and individual lines sorted by clusters.</p>
<div id="clustAnalyses" class="section level3">
<h3><span class="header-section-number">5.5.1</span> Analyses</h3>
<p>Clustering analyses were performed between individual lines based on the euclidean distance and the hierarchical clustering method implemented in the R <em>stats</em> package (using the <code>Callus</code> binary and <code>Root</code> count data; see section <a href="#Rclustroot">5.5.2</a> for R code). See Figure <a href="#fig:clusterTrees">5.1</a> for results and those are discussed below (by variable).</p>
<div class="figure"><span id="fig:clusterTrees"></span>
<img src="Analyses_rooting_files/figure-html/clusterTrees-1.png" alt="Clustering analyses based on callus (a) and root (b) data. For each variable, clusters are represented by shaded polygons, whereas the circles represent the assignment of the individual lines in the other analysis." width="100%" />
<p class="caption">
Figure 5.1: Clustering analyses based on callus (a) and root (b) data. For each variable, clusters are represented by shaded polygons, whereas the circles represent the assignment of the individual lines in the other analysis.
</p>
</div>
<p>To further compare the clusters identified in Figure <a href="#fig:clusterTrees">5.1</a> ridgeline plots (or density plots; more on this topic <a href="https://www.data-to-viz.com/graph/ridgeline.html">here</a>) were inferred for each variable using the R package <em>ggridges</em>. See Figure <a href="#fig:RidgelinesClusterTrees">5.2</a> for the results.</p>
<div class="figure"><span id="fig:RidgelinesClusterTrees"></span>
<img src="Analyses_rooting_files/figure-html/RidgelinesClusterTrees-1.png" alt="Ridgeline plots based on callus (a) and root (b) data. For each variable, individual lines are sorted based on the clustering analysis and ridgeline plots are colored based on clusters." width="100%" />
<p class="caption">
Figure 5.2: Ridgeline plots based on callus (a) and root (b) data. For each variable, individual lines are sorted based on the clustering analysis and ridgeline plots are colored based on clusters.
</p>
</div>
<div id="root-data" class="section level4">
<h4><span class="header-section-number">5.5.1.1</span> Root data</h4>
<p>This analysis sorted individuals into rooting clusters based on their abilities to initiate rooting (which is also shown by the ridgeline plot). Finally, to performer individuals (P1 and P2) identified by the statistical analyses are also represented on the graph. The code used to produce these analyses is available in section <a href="#Rclustroot">5.5.2</a>. The analysis clearly distinguished three clusters (of even number of individuals; see Table <a href="#tab:tablegroups">5.5</a>) based on individual rooting performances (Figure <a href="#fig:clusterTrees">5.1</a>):</p>
<ul>
<li>Black cluster: No to very little rooting.</li>
<li>Pink cluster: Some rooting observed.</li>
<li>Blue cluster: Most individuals are exhibiting high rooting capacity and the top three performers belong to this cluster.</li>
</ul>
<p>The number of individual per rooting clusters inferred from the clustering analysis is displayed in Table <a href="#tab:tablegroups">5.5</a>. The R code to generate this table is in section <a href="#Rclustroot">5.5.2</a>.</p>
<table>
<caption><span id="tab:tablegroups">Table 5.5: </span>Table summarizing composition of clusters based on rooting data.</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">G1</th>
<th align="right">G2</th>
<th align="right">TOT</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">black</td>
<td align="right">7</td>
<td align="right">8</td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="right">9</td>
<td align="right">8</td>
<td align="right">17</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="right">7</td>
<td align="right">6</td>
<td align="right">13</td>
</tr>
</tbody>
</table>
<div id="effect-of-growth-regulators-on-in-vitro-rooting-sorted-by-clusters" class="section level5">
<h5><span class="header-section-number">5.5.1.1.1</span> Effect of growth regulators on <em>in vitro</em> rooting sorted by clusters</h5>
<p>Table <a href="#tab:growthbyregulatorcluster">5.6</a> shows that IBA at both concentrations seem to have a significant effect on rooting for individuals belonging to the blue cluster (Figure <a href="#fig:RidgelinesClusterTrees">5.2</a>). Responses are very high (both at 87.18%) and the number of roots per tip are 2.97 +/- 2.25 for IBA 0.5 and 3.59 +/- 3.41 for IBA 1 (Table <a href="#tab:growthbyregulatorcluster">5.6</a>). The R code to generate this table is in section <a href="#Rclustroot">5.5.2</a>.</p>
<table>
<caption><span id="tab:growthbyregulatorcluster">Table 5.6: </span>Effect of growth regulator on in vitro rooting of Artemisia tridentata sorted by cluster. Response: mean of the three replicates +/- standard deviation; Av. No. of roots: mean +/- standard deviation.</caption>
<thead>
<tr class="header">
<th align="left">Cluster</th>
<th align="left">Growth regulator</th>
<th align="left">Concentration (mg/l)</th>
<th align="left">Response (%)</th>
<th align="left">Av. No. of roots</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">black</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="left">0 +/- 0</td>
<td align="left">0 +/- 0</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="left">11.11 +/- 0.32</td>
<td align="left">0.22 +/- 0.74</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="left">20 +/- 0.4</td>
<td align="left">0.44 +/- 1.27</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="left">13.33 +/- 0.34</td>
<td align="left">0.16 +/- 0.42</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="left">13.33 +/- 0.34</td>
<td align="left">0.22 +/- 0.67</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="left">30.77 +/- 0.47</td>
<td align="left">0.62 +/- 1.18</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="left">87.18 +/- 0.34</td>
<td align="left">2.97 +/- 2.25</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="left">87.18 +/- 0.34</td>
<td align="left">3.59 +/- 3.41</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="left">74.36 +/- 0.44</td>
<td align="left">1.74 +/- 1.96</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="left">66.67 +/- 0.48</td>
<td align="left">2.28 +/- 2.46</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="left">0 +/- 0</td>
<td align="left">0 +/- 0</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="left">49.02 +/- 0.5</td>
<td align="left">1.16 +/- 1.6</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="left">74.51 +/- 0.44</td>
<td align="left">1.82 +/- 1.88</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="left">39.22 +/- 0.49</td>
<td align="left">0.63 +/- 0.92</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="left">43.14 +/- 0.5</td>
<td align="left">0.86 +/- 1.48</td>
</tr>
</tbody>
</table>
<p>This will be investigated by looking at top performer individuals identified by i) statistical analyses and ii) clustering analysis.</p>
</div>
</div>
<div id="callus-data" class="section level4">
<h4><span class="header-section-number">5.5.1.2</span> Callus data</h4>
<p>The code used to produce these analyses is available in section <a href="#Rclustroot">5.5.2</a>. The analysis clearly distinguished three clusters based on individual lines producing callus (Figure <a href="#fig:clusterTrees">5.1</a>):</p>
<ul>
<li>Red cluster: Very limited callus production.</li>
<li>Green cluster: Intermediate callus production.</li>
<li>Orange cluster: High callus production.</li>
</ul>
<p>The clusters from the rooting analysis are not recovered here (each callus cluster are composed by a mixture of clusters from the rooting analysis) (see Figure <a href="#fig:clusterTrees">5.1</a>. This suggests that calli production do not inhibit rooting.</p>
<div id="effect-of-growth-regulators-on-in-vitro-calli-production-in-artemisia-tridentata-sorted-by-rooting-clusters" class="section level5">
<h5><span class="header-section-number">5.5.1.2.1</span> Effect of growth regulators on <em>in vitro</em> calli production in <em>Artemisia tridentata</em> sorted by rooting clusters</h5>
<p>Table <a href="#tab:growthbyregulatorclustercallus">5.7</a> shows that IBA at both concentrations seem to have a significant effect on callus production for individuals belonging to the blue cluster (Figure <a href="#fig:clusterTrees">5.1</a>). The R code is available in section <a href="#Rclustroot">5.5.2</a>.</p>
<table>
<caption><span id="tab:growthbyregulatorclustercallus">Table 5.7: </span>Effect of growth regulators on in vitro calli production on shoot tips of diploid Artemisia tridentata subsp. tridentata sorted by rooting clusters. Response: mean of the three replicates.</caption>
<thead>
<tr class="header">
<th align="left">Cluster</th>
<th align="left">Growth regulator</th>
<th align="left">Concentration (mg/l)</th>
<th align="right">Response (%)</th>
<th align="left">Treatment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">black</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">0.00</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">66.67</td>
<td align="left">IBA_05</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">68.89</td>
<td align="left">IBA_1</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">77.78</td>
<td align="left">NAA_05</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">82.22</td>
<td align="left">NAA_1</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">5.13</td>
<td align="left">Control</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">84.62</td>
<td align="left">IBA_05</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">97.44</td>
<td align="left">IBA_1</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">92.31</td>
<td align="left">NAA_05</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">89.74</td>
<td align="left">NAA_1</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">3.92</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">76.47</td>
<td align="left">IBA_05</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">88.24</td>
<td align="left">IBA_1</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">76.47</td>
<td align="left">NAA_05</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">90.20</td>
<td align="left">NAA_1</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="Rclustroot" class="section level3">
<h3><span class="header-section-number">5.5.2</span> R code</h3>
<pre class="r"><code>####~~~~
#Clustering analyses
####~~~~

#Perform clustering analyses on callus (binary) and root (count) data
# based on euclidian distance

###
# Callus data
###
# Input data for clustering and heatmap
mat_callus &lt;- table(MERGE$IndGeno, MERGE$Callus)

#Clustering analysis
clustCal &lt;- hclust(dist(mat_callus))
#convert to phylo class
trCal &lt;- as.phylo(clustCal)

#Find MRCA of clusters
#Red clade
RedNode &lt;- getMRCA(trCal, c(&quot;G2_b5_1&quot;, &quot;G1_b7_1&quot;))
#Green node
GreenNode &lt;- getMRCA(trCal, c(&quot;G2_b16_1&quot;, &quot;G2_b17_1&quot;))
#Orange node
OrangeNode &lt;- getMRCA(trCal, c(&quot;G1_b10_1&quot;, &quot;G2_b7_1&quot;))


###
# Root data
###
# Input data for clustering and heatmap
mat_root &lt;- table(MERGE$IndGeno, MERGE$Root)

#Clustering analysis
clustRt &lt;- hclust(dist(mat_root))
#convert to phylo class
trRt &lt;- as.phylo(clustRt)

#Find MRCA of clusters
#Blue clade
blueNode &lt;- getMRCA(trRt, c(&quot;G2_b27_1&quot;, &quot;G1_b2_1&quot;))
#Black node
blackNode &lt;- getMRCA(trRt, c(&quot;G2_b25_1&quot;, &quot;G2_b26_1&quot;))
#Pink node
pinkNode &lt;- getMRCA(trRt, c(&quot;G2_18_1&quot;, &quot;G1_b19_1&quot;))

###~~~
#Matrix with clusters and tips
###~~~
#This will serve to plot and compare the clustering analyses
MatClustInd &lt;- read.csv(&quot;02_Processed_data/Rooting_clusters_individuals.csv&quot;)

###~~~
#Draw trees
###~~~

#Callus tree
trCalplot &lt;- ggtree(trCal, layout=&quot;circular&quot;, branch.length=&quot;none&quot;, color=&quot;black&quot;) + 
  geom_tiplab(aes(angle=angle), color=&#39;black&#39;, offset=.7, cex = 2.5) +
  geom_hilight(node=RedNode, fill=&quot;red&quot;, alpha=.6) +
  geom_hilight(node=GreenNode, fill=&quot;green&quot;, alpha=.6) +
  geom_hilight(node=OrangeNode, fill=&quot;orange&quot;, alpha=.6) +
  geom_tippoint(color = as.character(MatClustInd[,2]), size = 3) +
  xlim_tree(22)

#Root tree
trRtplot &lt;- ggtree(trRt, layout=&quot;circular&quot;, branch.length=&quot;none&quot;, color=&quot;black&quot;) + 
  geom_tiplab(aes(angle=angle), color=&#39;black&#39;, offset=.6, cex = 2.5) +
  geom_hilight(node=blackNode, fill=&quot;black&quot;, alpha=.6) +
  geom_hilight(node=pinkNode, fill=&quot;pink&quot;, alpha=.6) +
  geom_hilight(node=blueNode, fill=&quot;blue&quot;, alpha=.6) +
  geom_tippoint(color = as.character(MatClustInd[,3]), size = 3) +
  xlim_tree(12)

ggarrange(trCalplot, trRtplot,ncol = 2, nrow = 1, labels=&quot;auto&quot;)

###~~~~
# Ridgeline plots
###~~~~
###~~~
#Add a new col in MERGE with order of tips based on hclust analysis
# This will help sorting individuals for ridgeline plot
###~~~

#Root data
is_tipRt &lt;- trRt$edge[,2] &lt;= length(trRt$tip.label)
ordered_tipsRt &lt;- trRt$edge[is_tipRt, 2]
newtipsRt &lt;- trRt$tip.label[ordered_tipsRt]

MERGE$ClustOrdRt &lt;- rep(&quot;NA&quot;, nrow(MERGE))
for(i in 1:length(newtipsRt)){
  MERGE$ClustOrdRt[grep(newtipsRt[i], MERGE$IndGeno)] &lt;- i
}
#Pad tip numbers to allow sorting them (for ridgeline plot)
MERGE$ClustOrdRt &lt;- str_pad(MERGE$ClustOrdRt, 2, pad = &quot;0&quot;)

#Callus data
is_tipCal &lt;- trCal$edge[,2] &lt;= length(trCal$tip.label)
ordered_tipsCal &lt;- trCal$edge[is_tipCal, 2]
newtipsCal &lt;- trCal$tip.label[ordered_tipsCal]

MERGE$ClustOrdCal &lt;- rep(&quot;NA&quot;, nrow(MERGE))
for(i in 1:length(newtipsCal)){
  MERGE$ClustOrdCal[grep(newtipsCal[i], MERGE$IndGeno)] &lt;- i
}
#Pad tip numbers to allow sorting them (for ridgeline plot)
MERGE$ClustOrdCal &lt;- str_pad(MERGE$ClustOrdCal, 2, pad = &quot;0&quot;)

###~~~
#Add clusters in MERGE: For coloring ridgeline series
###~~~
colInd &lt;- read.csv(&quot;02_Processed_data/Rooting_clusters_individuals.csv&quot;)
#Add col in MERGE with cluster
MERGE$ClusterRoot &lt;- colInd$Root_Cluster[match(MERGE$IndGeno, colInd$Individual)]
MERGE$ClusterRoot &lt;- as.vector(MERGE$ClusterRoot)

MERGE$ClusterCallus &lt;- colInd$Callus_Cluster[match(MERGE$IndGeno, colInd$Individual)]
MERGE$ClusterCallus &lt;- as.vector(MERGE$ClusterCallus)

###
#Ridgeline plot
###

# Callus ridgeline
CallusRidge &lt;- MERGE %&gt;%
  mutate(text = fct_reorder(IndGeno, ClustOrdCal)) %&gt;%
  ggplot(aes(x = Callus, y = text, fill = ClusterCallus)) +
  scale_fill_manual(values = unique(as.vector(MERGE$ClusterCallus))) +
  geom_density_ridges(stat=&quot;binline&quot;, binwidth = 1) +
  xlab(&quot;Callus data (0 = absence; 1 = presence)&quot;) +
  ylab(&quot;Individual lines (by clusters)     &quot;) +
  theme_ridges() + 
  theme(legend.position = &quot;none&quot;, text = element_text(size = 7), axis.text.y = element_text(size = 7)) 


# Root ridgeline
RootRidge &lt;- MERGE %&gt;%
  mutate(text = fct_reorder(IndGeno, ClustOrdRt)) %&gt;%
  ggplot(aes(x = Root, y = text, fill = ClusterRoot)) +
  scale_fill_manual(values = c(&quot;grey&quot;, &quot;blue&quot;,&quot;pink&quot;)) +
  geom_density_ridges(stat=&quot;binline&quot;, binwidth = 1) +
  theme_ridges() + 
  xlab(&quot;Root data (number of roots per shoot tip)&quot;) +
  ylab(&quot;Individual lines (by clusters)     &quot;) +
  theme(legend.position = &quot;none&quot;, text = element_text(size = 7), axis.text.y = element_text(size = 7)) 

ggarrange(CallusRidge, RootRidge, ncol = 2, nrow = 1, labels=&quot;auto&quot;)

###~~~
#Produce pivot table with number of individual per group and genotype composition
###~~~
# Load file with color of individuals (which group based on clusters they belong to)
colInd &lt;- read.csv(&quot;02_Processed_data/Rooting_clusters_individuals.csv&quot;)

###~~~
#Effect of growth regulators on *in vitro* rooting sorted by clusters
###~~~
# colInd add Genotypes
colInd$Genotype &lt;- sapply(strsplit(as.vector(colInd$Individual), split=&quot;_&quot;), &quot;[[&quot;, 1)
tabGroup &lt;- table(colInd$Group, colInd$Genotype)
tabGroup &lt;- tabGroup[c(1,3,2),]
tabGroup &lt;- as.data.frame.matrix(tabGroup)
tabGroup$TOT &lt;- rowSums(tabGroup)

# Plot table
knitr::kable(tabGroup, row.names=T, caption = &quot;Table summarizing composition of clusters based on rooting data.&quot;)

#Load file with color of individuals (which group based on clusters they belong to)
colInd &lt;- read.csv(&quot;02_Processed_data/Rooting_clusters_individuals.csv&quot;)
treat &lt;- sort(unique(MERGE$Treatment))

#Matrix summarizing effect of growth regulators on rooting
RootingStatbyTreatC &lt;- matrix(ncol=6, nrow=15)
colnames(RootingStatbyTreatC) &lt;- c(&quot;Cluster&quot;,&quot;Growth regulator&quot;, &quot;Concentration (mg/l)&quot;, &quot;Response (%)&quot;, &quot;Av. No. of roots&quot;, &quot;Treatment&quot;)

#Add col in MERGE with cluster
MERGE$ClusterRoot &lt;- colInd$Group[match(MERGE$IndGeno, colInd$Individual)]

#Convert to dataframe
RootingStatbyTreatC &lt;- as.data.frame(RootingStatbyTreatC)

#Populate matrix
RootingStatbyTreatC$Cluster &lt;- sort(rep(sort(unique(MERGE$ClusterRoot)),5))
RootingStatbyTreatC$`Growth regulator` &lt;- rep(sapply(strsplit(treat, split=&quot;_&quot;),&quot;[[&quot;,1),3)
RootingStatbyTreatC$`Concentration (mg/l)` &lt;- rep(c(&quot;-&quot;, sapply(strsplit(treat[2:length(treat)], split=&quot;_&quot;),&quot;[[&quot;,2)),3)
RootingStatbyTreatC$Treatment &lt;- rep(treat, 3)

#Infer Response and mean no roots
for(i in 1:nrow(RootingStatbyTreatC)){
  foo &lt;- subset(MERGE, MERGE$Treatment == RootingStatbyTreatC$Treatment[i] &amp; MERGE$ClusterRoot == RootingStatbyTreatC$Cluster[i])
  RootingStatbyTreatC$`Response (%)`[i] &lt;- paste(round(100*(mean(foo$RootBin)),2), &quot;+/-&quot;, round(sd(foo$RootBin),2), sep= &quot; &quot;)
  RootingStatbyTreatC$`Av. No. of roots`[i] &lt;- paste(round(mean(foo$Root),2), &quot;+/-&quot;, round(sd(foo$Root),2), sep=&quot; &quot;)
}

#Plot table
knitr::kable(RootingStatbyTreatC[,1:5], row.names=F, caption = &quot;Effect of growth regulator on in vitro rooting of Artemisia tridentata sorted by cluster. Response: mean of the three replicates +/- standard deviation; Av. No. of roots: mean +/- standard deviation.&quot;)

###~~~
#Effect of growth regulators on calli production sorted by rooting cluster
###~~~

# Load file with color of individuals (which group based on clusters they belong to)
colInd &lt;- read.csv(&quot;02_Processed_data/Rooting_clusters_individuals.csv&quot;)

# Create vector of treatments
treat &lt;- sort(unique(MERGE$Treatment))

# Matrix summarizing effect of growth regulators on rooting
RootingStatbyTreatCal &lt;- matrix(ncol=5, nrow=15)
colnames(RootingStatbyTreatCal) &lt;- c(&quot;Cluster&quot;,&quot;Growth regulator&quot;, &quot;Concentration (mg/l)&quot;, &quot;Response (%)&quot;, &quot;Treatment&quot;)

# Add col in MERGE with cluster
MERGE$ClusterRoot &lt;- colInd$Group[match(MERGE$IndGeno, colInd$Individual)]

# Convert to dataframe
RootingStatbyTreatCal &lt;- as.data.frame(RootingStatbyTreatCal)

# Populate matrix
RootingStatbyTreatCal$Cluster &lt;- sort(rep(sort(unique(MERGE$ClusterRoot)),5))
RootingStatbyTreatCal$`Growth regulator` &lt;- rep(sapply(strsplit(treat, split=&quot;_&quot;),&quot;[[&quot;,1),3)
RootingStatbyTreatCal$`Concentration (mg/l)` &lt;- rep(c(&quot;-&quot;, sapply(strsplit(treat[2:length(treat)], split=&quot;_&quot;),&quot;[[&quot;,2)),3)
RootingStatbyTreatCal$Treatment &lt;- rep(treat, 3)

# Infer Response
for(i in 1:nrow(RootingStatbyTreatCal)){
  foo &lt;- subset(MERGE, MERGE$Treatment == RootingStatbyTreatCal$Treatment[i] &amp; MERGE$ClusterRoot == RootingStatbyTreatCal$Cluster[i])
  RootingStatbyTreatCal$`Response (%)`[i] &lt;- round(100*(mean(foo$Callus)),2)
}

#Plot table
knitr::kable(RootingStatbyTreatCal[,1:5], row.names=F, caption = &quot;Effect of growth regulators on in vitro calli production on shoot tips of diploid Artemisia tridentata subsp. tridentata sorted by rooting clusters. Response: mean of the three replicates.&quot;)</code></pre>
</div>
</div>
</div>
<div id="step4" class="section level1">
<h1><span class="header-section-number">6</span> Step 4: <em>In vitro</em> survival and growth of plantlets</h1>
<div id="sampling-and-preliminary-analyses" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">6.1</span> Sampling and preliminary analyses</h2>
<div id="analyses-7" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Analyses</h3>
<p>Data on survival and plantlets heights are stored in <code>01_Raw_Data/Survival_height_clones.csv</code>. The code below provides statistics on the number of individual lines and shoot tips included in this experiment. The R code for these analyses is available in section <a href="#codesurv">6.1.2</a>.</p>
<pre><code>##                        Sampling       
## Total N. ind.          &quot;42&quot;           
## N. ind. per gen,       &quot;G1:20/G2:22&quot;  
## Total N. shoot tips    &quot;273&quot;          
## N. shoot tips per gen. &quot;G1:138/G2:135&quot;</code></pre>
<pre><code>## 
## G1_b13_1  G1_b3_1 G2_b27_1  G2_b7_1  G1_b8_1 G2_b24_1 G2_b21_1  G2_b4_1 
##       12       12       12       12       11       11       10       10 
## G1_b10_1  G1_b2_1 G1_b29_1 G2_b11_1 G1_b16_1 G1_b25_1  G2_b8_1  G1_b1_2 
##        9        9        9        9        8        8        8        7 
## G1_b21_1  G1_b5_1 G2_b16_1 G2_b20_1 G2_b21_2 G1_b20_1  G1_b4_1  G1_b9_1 
##        7        7        7        7        7        6        6        6 
##  G2_29_1 G2_b17_1 G2_b27_2 G1_b19_1  G2_18_1  G1_b1_1 G1_b18_1  G1_b6_1 
##        6        6        6        5        5        4        4        4 
## G2_b15_1  G2_b2_1 G1_b26_1 G2_b28_1  G2_b6_1  G2_b5_1 G1_b15_1 G2_b19_1 
##        4        4        3        3        3        2        1        1 
## G2_b25_1 G2_b26_1 
##        1        1</code></pre>
<div id="survival-and-plantlets-heights" class="section level4">
<h4><span class="header-section-number">6.1.1.1</span> Survival and plantlets heights</h4>
<p>Results of the code provided below is summarized in Table <a href="#tab:surv">6.1</a>. After 5 weeks, a survival of 53.48% of the plantlets was observed. A significant increase of plantlets mortality has been recorded between weeks 3 and 5 and on average, plantlets did not growth significant (but see below for more details).</p>
<table>
<caption><span id="tab:surv">Table 6.1: </span>Survival and plantlets heights at 3 and 5 weeks (+/- standard deviations).</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Survival (%)</th>
<th align="left">Plant height (cm)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Week 3</td>
<td align="left">75.82</td>
<td align="left">1.78+/-1.21</td>
</tr>
<tr class="even">
<td align="left">Week 5</td>
<td align="left">53.48</td>
<td align="left">1.69+/-1.65</td>
</tr>
</tbody>
</table>
</div>
<div id="survival-and-plantlets-heights-by-rooting-clusters" class="section level4">
<h4><span class="header-section-number">6.1.1.2</span> Survival and plantlets heights by rooting clusters</h4>
<p>Table <a href="#tab:survclust">6.2</a> shows data on survival, plantlets heights at 3 and 5 weeks sorted by rooting clusters. Data on the number of individuals and shoot tips included in this experiment for each rooting cluster is also provided.</p>
<table style="width:100%;">
<caption><span id="tab:survclust">Table 6.2: </span>Survival and plantlets heights at 3 and 5 weeks sorted by rooting clusters.</caption>
<colgroup>
<col width="14%" />
<col width="7%" />
<col width="12%" />
<col width="13%" />
<col width="20%" />
<col width="13%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Rooting cluster</th>
<th align="left">N. ind.</th>
<th align="left">N. shoot tips</th>
<th align="left">3 w. surv. (%)</th>
<th align="left">3 w. plant height (cm)</th>
<th align="left">5 w. surv. (%)</th>
<th align="left">5 w. plant height (cm)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">black</td>
<td align="left">12</td>
<td align="left">28</td>
<td align="left">71.43</td>
<td align="left">1.49+/-1.13</td>
<td align="left">42.86</td>
<td align="left">1.36+/-1.69</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="left">16</td>
<td align="left">104</td>
<td align="left">70.19</td>
<td align="left">1.5+/-1.13</td>
<td align="left">50</td>
<td align="left">1.48+/-1.55</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="left">13</td>
<td align="left">134</td>
<td align="left">83.58</td>
<td align="left">2.09+/-1.18</td>
<td align="left">59.7</td>
<td align="left">1.95+/-1.67</td>
</tr>
</tbody>
</table>
</div>
<div id="survival-and-plantlets-heights-by-rooting-clusters-and-treatments" class="section level4">
<h4><span class="header-section-number">6.1.1.3</span> Survival and plantlets heights by rooting clusters and treatments</h4>
<p>Table <a href="#tab:survclusttreat">6.3</a> shows survival and plantlets heights at 5 weeks sorted by rooting clusters and treatments.</p>
<table>
<caption><span id="tab:survclusttreat">Table 6.3: </span>Survival and plantlets heights at 5 weeks sorted by rooting clusters and treatments.</caption>
<colgroup>
<col width="7%" />
<col width="7%" />
<col width="13%" />
<col width="16%" />
<col width="20%" />
<col width="12%" />
<col width="11%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Cluster</th>
<th align="right">N. ind.</th>
<th align="right">N. shoot tips</th>
<th align="left">Growth regulator</th>
<th align="left">Concentration (mg/l)</th>
<th align="right">Survival (%)</th>
<th align="left">Height (cm)</th>
<th align="left">Treatment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">black</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">0.00</td>
<td align="left">0+/-NA</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">40.00</td>
<td align="left">1.2+/-1.64</td>
<td align="left">IBA_05</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="right">4</td>
<td align="right">6</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">33.33</td>
<td align="left">0.9+/-1.59</td>
<td align="left">IBA_1</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="right">4</td>
<td align="right">6</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">66.67</td>
<td align="left">2.27+/-1.94</td>
<td align="left">NAA_05</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="right">8</td>
<td align="right">10</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">40.00</td>
<td align="left">1.32+/-1.71</td>
<td align="left">NAA_1</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="right">8</td>
<td align="right">12</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">58.33</td>
<td align="left">1.97+/-1.8</td>
<td align="left">Control</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="right">13</td>
<td align="right">32</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">65.62</td>
<td align="left">2.27+/-1.7</td>
<td align="left">IBA_05</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="right">13</td>
<td align="right">34</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">64.71</td>
<td align="left">1.95+/-1.54</td>
<td align="left">IBA_1</td>
</tr>
<tr class="odd">
<td align="left">blue</td>
<td align="right">12</td>
<td align="right">27</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">37.04</td>
<td align="left">1.21+/-1.64</td>
<td align="left">NAA_05</td>
</tr>
<tr class="even">
<td align="left">blue</td>
<td align="right">13</td>
<td align="right">29</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">68.97</td>
<td align="left">2.29+/-1.64</td>
<td align="left">NAA_1</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Control</td>
<td align="left">-</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="right">14</td>
<td align="right">24</td>
<td align="left">IBA</td>
<td align="left">05</td>
<td align="right">37.50</td>
<td align="left">1.04+/-1.39</td>
<td align="left">IBA_05</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="right">16</td>
<td align="right">35</td>
<td align="left">IBA</td>
<td align="left">1</td>
<td align="right">60.00</td>
<td align="left">1.82+/-1.59</td>
<td align="left">IBA_1</td>
</tr>
<tr class="even">
<td align="left">pink</td>
<td align="right">13</td>
<td align="right">19</td>
<td align="left">NAA</td>
<td align="left">05</td>
<td align="right">42.11</td>
<td align="left">1.28+/-1.59</td>
<td align="left">NAA_05</td>
</tr>
<tr class="odd">
<td align="left">pink</td>
<td align="right">15</td>
<td align="right">26</td>
<td align="left">NAA</td>
<td align="left">1</td>
<td align="right">53.85</td>
<td align="left">1.56+/-1.57</td>
<td align="left">NAA_1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="codesurv" class="section level3">
<h3><span class="header-section-number">6.1.2</span> R code</h3>
<pre class="r"><code>###~~~
#Load data on survival and height
###~~~
survheig &lt;- read.csv(&quot;01_Raw_Data/Survival_height_clones.csv&quot;)

###~~~
#How many individuals where transfered corresponding how many shoot tips
###~~~
stats &lt;- matrix(ncol=1, nrow=4)
colnames(stats) &lt;- &quot;Sampling&quot;
rownames(stats) &lt;- c(&quot;Total N. ind.&quot;, &quot;N. ind. per gen,&quot;,&quot;Total N. shoot tips&quot;, &quot;N. shoot tips per gen.&quot;)

#Add N ind and N shoot tips
stats[c(1,3),] &lt;- c(length(unique(survheig$SeedID)), sum(table(survheig$SeedID)))

#Tot ind per genotypes
totindgen &lt;- table(sapply(strsplit(as.vector(unique(survheig$SeedID)), &quot;_&quot;), &#39;[[&#39;, 1))
stats[2,1] &lt;- paste0(paste0(names(totindgen), &quot;:&quot;), totindgen, collapse = &#39;/&#39;)

#N shoot per genotypes
totgenshoot &lt;- table(sapply(strsplit(as.vector(survheig$SeedID), &quot;_&quot;), &#39;[[&#39;, 1))
stats[4,1] &lt;- paste0(paste0(names(totgenshoot), &quot;:&quot;), totgenshoot, collapse = &#39;/&#39;)


print(stats)

###~~~
#Breakdown of number of shoot tips per individual
###~~~

#Sort based on number of shoot tips
sort(table(survheig$SeedID), decreasing=T)

###~~~
# Matrix to store data @ 3 and 5 weeks
###~~~
survheigStat &lt;- matrix(ncol=2, nrow = 2)
colnames(survheigStat) &lt;- c(&quot;Survival (%)&quot;, &quot;Plant height (cm)&quot;)
rownames(survheigStat) &lt;- c(&quot;Week 3&quot;, &quot;Week 5&quot;)

###
#3 weeks
###

# Survivial
ThreeWsurvAll &lt;- table(survheig$X3_w_survival)
survheigStat[1,1] &lt;- round(100*((sum(ThreeWsurvAll)-ThreeWsurvAll[which(names(ThreeWsurvAll) == 0)])/sum(ThreeWsurvAll)),2)

# Mean plant height +/- sd
survheigStat[1,2] &lt;- paste0(round(mean(survheig$X3_w_height),2), &quot;+/-&quot;, round(sd(survheig$X3_w_height),2))

###
#5 weeks
###

# Survivial
FiveWsurvAll &lt;- table(survheig$X5_w_survival)
survheigStat[2,1] &lt;- round(100*((sum(FiveWsurvAll)-FiveWsurvAll[which(names(FiveWsurvAll) == 0)])/sum(FiveWsurvAll)),2)

# Mean plant height +/- sd
survheigStat[2,2] &lt;- paste0(round(mean(survheig$X5_w_height),2), &quot;+/-&quot;, round(sd(survheig$X5_w_height),2))


# Plot table
knitr::kable(survheigStat, row.names=T, caption = &quot;Survival and plantlets heights at 3 and 5 weeks (+/- standard deviations).&quot;)

###
#Create list of clusters
###
clust &lt;- sort(unique(survheig$Cluster))[c(1,3,2)]

###~~~
#Matrix with data on cluster, N. ind, N. shoot tips, survival and height
###~~~
clustSurvHeig &lt;- matrix(ncol=7, nrow = length(clust))
colnames(clustSurvHeig) &lt;- c(&quot;Rooting cluster&quot;, &quot;N. ind.&quot;,&quot;N. shoot tips&quot;, &quot;3 w. surv. (%)&quot;, &quot;3 w. plant height (cm)&quot;, &quot;5 w. surv. (%)&quot;, &quot;5 w. plant height (cm)&quot;)
clustSurvHeig[,1] &lt;- as.character(clust)

for(i in 1:length(clust)){
  #Subset mat to clust
  tmp &lt;- subset(survheig, survheig$Cluster == clust[i])
  
  ###
  #3 weeks
  ###
  #N. ind.
  clustSurvHeig[i,2] &lt;- length(unique(tmp$SeedID))
  #N. shoot tips
  clustSurvHeig[i,3] &lt;- nrow(tmp)
  #Survivial
  ThreeWsurv &lt;- table(tmp$X3_w_survival)
  clustSurvHeig[i,4] &lt;- round(100*((sum(ThreeWsurv)-ThreeWsurv[which(names(ThreeWsurv) == 0)])/sum(ThreeWsurv)),2)
  #Mean plant height +/- sd
  clustSurvHeig[i,5] &lt;- paste0(round(mean(tmp$X3_w_height),2), &quot;+/-&quot;, round(sd(tmp$X3_w_height),2))
  
  ###
  #5 weeks
  ###
  #Survivial
  FiveWsurv &lt;- table(tmp$X5_w_survival)
  clustSurvHeig[i,6] &lt;- round(100*((sum(FiveWsurv)-FiveWsurv[which(names(FiveWsurv) == 0)])/sum(FiveWsurv)),2)
  #Mean plant height +/- sd
  clustSurvHeig[i,7] &lt;- paste0(round(mean(tmp$X5_w_height),2), &quot;+/-&quot;, round(sd(tmp$X5_w_height),2))
}

#Plot table
knitr::kable(clustSurvHeig, row.names=F, caption = &quot;Survival and plantlets heights at 3 and 5 weeks sorted by rooting clusters.&quot;)

###~~~
#Split survheig$Treatment to extract growth regulator and concentration
###~~~
survheig$TreatConc &lt;- paste(sapply(strsplit(as.vector(survheig$Treatment), split=&quot;_&quot;), &quot;[[&quot;, 2), sapply(strsplit(as.vector(survheig$Treatment), split=&quot;_&quot;), &quot;[[&quot;, 3), sep=&quot;_&quot;)
survheig$TreatConc[grep(&quot;Control_&quot;, survheig$TreatConc)] &lt;- &quot;Control&quot;

#Vector of treatments
treat &lt;- sort(unique(survheig$TreatConc))

###~~~
#Matrix summarizing effect of growth regulators on rooting
###~~~
survheigTreat &lt;- matrix(ncol=8, nrow=15)
colnames(survheigTreat) &lt;- c(&quot;Cluster&quot;, &quot;N. ind.&quot;,&quot;N. shoot tips&quot;,&quot;Growth regulator&quot;, &quot;Concentration (mg/l)&quot;, &quot;Survival (%)&quot;, &quot;Height (cm)&quot;, &quot;Treatment&quot;)

#Convert to dataframe
survheigTreat &lt;- as.data.frame(survheigTreat)

#Populate matrix
survheigTreat$Cluster &lt;- sort(rep(clust,5))
survheigTreat$`Growth regulator` &lt;- rep(sapply(strsplit(treat, split=&quot;_&quot;),&quot;[[&quot;,1),3)
survheigTreat$`Concentration (mg/l)` &lt;- rep(c(&quot;-&quot;, sapply(strsplit(treat[2:length(treat)], split=&quot;_&quot;),&quot;[[&quot;,2)),3)
survheigTreat$Treatment &lt;- rep(treat, 3)

#Infer N, surv and plant height
for(i in 1:nrow(survheigTreat)){
  foo &lt;- subset(survheig, survheig$TreatConc == survheigTreat$Treatment[i] &amp; survheig$Cluster == survheigTreat$Cluster[i])
  
  if(nrow(foo) &gt; 0){
    #N. ind.
    survheigTreat$`N. ind.`[i] &lt;- length(unique(foo$SeedID))
    #N. shoot tips
    survheigTreat$`N. shoot tips`[i] &lt;- nrow(foo)
    #Survivial
    ThreeWsurv &lt;- table(foo$X5_w_survival)
    survheigTreat$`Survival (%)`[i] &lt;-   round(100*((sum(ThreeWsurv)-ThreeWsurv[which(names(ThreeWsurv) == 0)])/sum(ThreeWsurv)),2)
    #Mean plant height +/- sd
    survheigTreat$`Height (cm)`[i] &lt;- paste0(round(mean(foo$X5_w_height),2), &quot;+/-&quot;, round(sd(foo$X5_w_height),2))  
  }
  
}

#Save to be included in manuscript
#write.csv(survheigTreat, file=&#39;Survival_cluster_treatment_5w.csv&#39;)

#Plot table
knitr::kable(survheigTreat, row.names=F, caption = &quot;Survival and plantlets heights at 5 weeks sorted by rooting clusters and treatments.&quot;)</code></pre>
</div>
</div>
<div id="statistical-analyses-on-survival-data" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">6.2</span> Statistical analyses on survival data</h2>
<div id="analyses-8" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Analyses</h3>
<p>Here, we will use statistical analyses to test whether treatment or individual predict survival and heights of plantlets after 5-weeks of culture. The distribution of the survival data is in Figure <a href="#fig:distrSurv">6.1</a> and it supports using GLM models with a binomial family distribution (same approach than with the callus data). The R code is available in section <a href="#survcodeR">6.2.2</a>.</p>
<div class="figure"><span id="fig:distrSurv"></span>
<img src="Analyses_rooting_files/figure-html/distrSurv-1.png" alt="Barplot of survival of plantlets after 5 weeks of growth." width="672" />
<p class="caption">
Figure 6.1: Barplot of survival of plantlets after 5 weeks of growth.
</p>
</div>
<div id="treatment-effect-on-survival" class="section level4">
<h4><span class="header-section-number">6.2.1.1</span> Treatment effect on survival</h4>
<p>We are using Fisher tests to test whether treatments had an effect on survivals of seedlings.</p>
<pre><code>##      control IBA_05 IBA_1 NAA_05 NAA_1
## live       6     29    45     19    38
## dead       6     12    28     40    25</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  survival
## p-value = 0.0008326
## alternative hypothesis: two.sided</code></pre>
<p>The analysis showed that there is a treatment effect on survival.</p>
</div>
</div>
<div id="survcodeR" class="section level3">
<h3><span class="header-section-number">6.2.2</span> R code</h3>
<pre class="r"><code>####~~~~
#Infer treatment types recieved by plantles
###~~~
survheig$Treat &lt;- paste(sapply(strsplit(as.vector(survheig$Treatment), split=&quot;_&quot;), &quot;[[&quot;, c(2)), sapply(strsplit(as.vector(survheig$Treatment), split=&quot;_&quot;), &quot;[[&quot;, c(3)), sep=&quot;_&quot;)
survheig$Treat[grep(&quot;Control_&quot;, survheig$Treat)] &lt;- &quot;Control&quot;

###~~~
#Turn survival into a binary data (sickly plantlets have a score of 0.5, but will be changed to 1)
###~~~
survheig$X5_w_survival[which(survheig$X5_w_survival == 0.5)] &lt;- 1

###~~~
#Produce barplot of survival data
###~~~
barplot(table(survheig$X5_w_survival), xlab=&quot;Binary data on survival&quot;, ylab=&quot;Occurrence&quot;)

###~~~
#Treatment effect on survival
##~~~
survival &lt;- matrix(c(6, 6, 29, 12, 45, 28, 19, 40, 38, 25 ), nr=2)
colnames(survival) = c(&quot;control&quot;, &quot;IBA_05&quot;,&quot;IBA_1&quot;,&quot;NAA_05&quot;,&quot;NAA_1&quot;)
row.names(survival) = c(&quot;live&quot;, &quot;dead&quot;)
print (survival)

###~~~
#Fisher test
###~~~
fisher.test(survival)</code></pre>
</div>
</div>
<div id="statistical-analyses-on-growth-data" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">6.3</span> Statistical analyses on growth data</h2>
<p>Here, we will only work on alive plantlets (<code>survheig$X5_w_survival</code> == 1) and conduct statistical analyses to test whether individuals from the different rooting clusters have different growth rates. First, we have to test whether the continuous growth data follow a normal distribution using the Shapiro-Wilk normality test. The R code for all these analyses is available in section <a href="#codegrowthstat">6.3.2</a>.</p>
<div id="analyses-9" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Analyses</h3>
<div id="test-for-normality" class="section level4">
<h4><span class="header-section-number">6.3.1.1</span> Test for normality</h4>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  subSurvHeigh$X5_w_height
## W = 0.97501, p-value = 0.009041</code></pre>
</div>
<div id="anova-analyses" class="section level4">
<h4><span class="header-section-number">6.3.1.2</span> ANOVA analyses</h4>
<pre><code>##              Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## Cluster       2  3.195  1.5977   6.224  0.00276 ** 
## SeedID       32 24.823  0.7757   3.022 1.02e-05 ***
## Residuals   109 27.979  0.2567                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 2 observations deleted due to missingness</code></pre>
<p>The ANOVA show significant cluster and individual effects. The Tukey tests show that individuals in the blue rooting cluster are significantly taller than those in the pink cluster (black is not shown different than blue, but it could be due to small sampling size of this cluster). Finally, <code>G2_b27_1</code> is significantly taller than 13.88% of the individuals included in this experiment. This individual was identified as a top 3 performer!</p>
</div>
<div id="tukeys-tests" class="section level4">
<h4><span class="header-section-number">6.3.1.3</span> Tukey’s tests</h4>
<pre><code>##                   diff        lwr        upr       p adj
## blue-black  0.08416667 -0.2727044  0.4410378 0.834669945
## pink-black -0.23333333 -0.6025240  0.1358574 0.284234864
## pink-blue  -0.31750000 -0.5228494 -0.1121506 0.001530769</code></pre>
<pre><code>## [1] &quot;Blue is significantly outperforming pink!&quot;</code></pre>
<pre><code>## G2_b27_1 
## 13.88889</code></pre>
</div>
</div>
<div id="codegrowthstat" class="section level3">
<h3><span class="header-section-number">6.3.2</span> R code</h3>
<pre class="r"><code>###~~~
#Subset surveigh to include only alive plantlets after 5 weeks
###~~~

#All plantlets surviving after 5 weeks of culture 
subSurvHeigh &lt;- subset(survheig, survheig$X5_w_survival != &quot;0&quot;)

###~~~
#Test for normality
###~~~
# The data have a normal distribution!
shapiro.test(subSurvHeigh$X5_w_height)

###~~~
#Define ANOVA model and run analysis
###~~~
#Since it seemed that treatment had an effect
anovaOUTRoot &lt;- aov(X5_w_height ~ Cluster + SeedID, data = subSurvHeigh)


# Summary of ANOVA
summary(anovaOUTRoot)

###~~~
#Tukey tests on significant variables
###~~~
# Vector with significant variables (only one here)
significant_vars &lt;- c(&quot;SeedID&quot;, &quot;Cluster&quot;)

# Run Tukey tests
anovaOUTheight_tukey &lt;- TukeyHSD(anovaOUTRoot, significant_vars)

# Save analysis
saveRDS(anovaOUTheight_tukey,&quot;02_Processed_data/rds_tables/TukeyHeight.rds&quot;)

###~~~
#Investigate cluster effect (**)
###~~~
ClustTukeyHeight &lt;- anovaOUTheight_tukey$Cluster[which(as.numeric(anovaOUTheight_tukey$Cluster[,4]) &lt;= 0.01),]

# What are the top cluster(s) (= most efficient at growing fast)?
#ClusterTukeyHeightTop &lt;- ClustTukeyHeight[which(ClustTukeyHeight[,1] &gt; 0),]

print(anovaOUTheight_tukey$Cluster)
print(&quot;Blue is significantly outperforming pink!&quot;)

###~~~
#Investigate individual effect (***)
###~~~
IndTukeyHeight &lt;- anovaOUTheight_tukey$SeedID[which(as.numeric(anovaOUTheight_tukey$SeedID[,4]) &lt;= 0.01),]

# What are the top performers (= most efficient at growing fast)?
IndTukeyHeightTop &lt;- IndTukeyHeight[which(IndTukeyHeight[,1] &gt; 0),]

# Extract best performing ind (first in rownames of comparison)
P1growth &lt;- sapply(strsplit(rownames(IndTukeyHeightTop), split=&#39;-&#39;), &quot;[[&quot;,1)

# Percentage of individuals that are outperformed by P1
TopIndgrowth &lt;- 100*(sort(table(P1growth), decreasing=T)/length(unique(subSurvHeigh$SeedID)))

## This analysis shows that G2_b27_1 grows significantly faster than 14.28% of the individuals in this experiment
print(TopIndgrowth)</code></pre>
</div>
</div>
<div id="boxplot-of-plantlet-heights-for-individuals-in-blue-rooting-cluster" class="section level2 tabset tabset-pills">
<h2><span class="header-section-number">6.4</span> Boxplot of plantlet heights for individuals in blue rooting cluster</h2>
<div id="analyses-10" class="section level3">
<h3><span class="header-section-number">6.4.1</span> Analyses</h3>
<p>Figure <a href="#fig:boxplotheightblue">6.2</a> shows the distribution of plantlets heights after five weeks of culture for individuals belonging to the blue rooting cluster of diploid <em>Artemisia tridentata</em> subsp. <em>tridentata</em>. As predicted by the statistical analyses on plantlet heights, <code>G2_b7_1</code> among the top three individuals with the fastest growth in our experiment (with <code>G2_b7_1</code> being the fastest grower and isolated as P2 in previous analyses). The top two individuals identified by the rooting analysis are also shown on the boxplot. The R code is available in section <a href="#boxgrowthcode">6.4.2</a>.</p>
<div class="figure"><span id="fig:boxplotheightblue"></span>
<img src="Analyses_rooting_files/figure-html/boxplotheightblue-1.png" alt="Boxplot showing plantlets heights after five weeks of culture for individuals belonging to the blue rooting cluster of diploid Artemisia tridentata subsp. tridentata. The n indicates the number of plantlets cultured for each individual. The top three performers as identified by the rooting experiment are also displayed (P1, P2)." width="672" />
<p class="caption">
Figure 6.2: Boxplot showing plantlets heights after five weeks of culture for individuals belonging to the blue rooting cluster of diploid Artemisia tridentata subsp. tridentata. The n indicates the number of plantlets cultured for each individual. The top three performers as identified by the rooting experiment are also displayed (P1, P2).
</p>
</div>
</div>
<div id="boxgrowthcode" class="section level3">
<h3><span class="header-section-number">6.4.2</span> R code</h3>
<pre class="r"><code>###~~~
#Subset to only include from blue rooting cluster
###~~~
blueHeight &lt;- subset(subSurvHeigh, subSurvHeigh$Cluster == &#39;blue&#39;)

#Individual
seedID &lt;- table(as.vector(blueHeight$SeedID))

# Create a vector named &quot;new_order&quot; containing the desired order to display individuals (smaller to taller)
new_order &lt;- with(blueHeight, reorder(as.vector(blueHeight$SeedID), as.numeric(blueHeight$X5_w_height), median , na.rm=T))

###~~~
# Draw boxplot
###~~~
boxplot(as.numeric(blueHeight$X5_w_height) ~ new_order, names = paste(names(seedID), paste0(&quot;(n: &quot;, as.vector(seedID), &quot;)&quot;), sep = &quot;\n&quot;), las=2, xlab = &quot;&quot;, ylab = &quot;Plantlet height (cm)&quot;, cex = .8)

#Add P1, P2
top &lt;- which(names(seedID) %in% c(&quot;G2_b27_1&quot;, &quot;G2_b7_1&quot;))
text(x=top, y=1.8, c(&quot;P1&quot;, &quot;P2&quot;))</code></pre>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1><span class="header-section-number">7</span> References</h1>
<div id="refs">
<div id="ref-R-emmeans">
<p>Lenth, Russell. 2020. <em>Emmeans: Estimated Marginal Means, Aka Least-Squares Means</em>. <a href="https://github.com/rvlenth/emmeans">https://github.com/rvlenth/emmeans</a>.</p>
</div>
</div>
</div>
<div id="appendix-1" class="section level1 appendix">
<h1><span class="header-section-number">8</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.4. 2020. &lt;URL:
<a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.10. 2019. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] D. Ebbert. <em>chisq.posthoc.test: A Post Hoc Analysis for Pearson’s
Chi-Squared Test for Count Data</em>. R package version 0.1.2. 2019. &lt;URL:
<a href="http://chisq-posthoc-test.ebbert.nrw/" class="uri">http://chisq-posthoc-test.ebbert.nrw/</a>&gt;.</p>
<p>[4] T. Hothorn, A. Zeileis, R. W. Farebrother, et al. <em>lmtest: Testing
Linear Regression Models</em>. R package version 0.9-38. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=lmtest" class="uri">https://CRAN.R-project.org/package=lmtest</a>&gt;.</p>
<p>[5] S. Jackman, A. Tahk, A. Zeileis, et al. <em>pscl: Political Science
Computational Laboratory</em>. R package version 1.5.5. 2020. &lt;URL:
<a href="http://github.com/atahk/pscl" class="uri">http://github.com/atahk/pscl</a>&gt;.</p>
<p>[6] A. Kassambara. <em>ggpubr: ggplot2 Based Publication Ready Plots</em>. R
package version 0.4.0. 2020. &lt;URL:
<a href="https://rpkgs.datanovia.com/ggpubr/" class="uri">https://rpkgs.datanovia.com/ggpubr/</a>&gt;.</p>
<p>[7] R. Lenth. <em>emmeans: Estimated Marginal Means, aka Least-Squares
Means</em>. R package version 1.5.2-1. 2020. &lt;URL:
<a href="https://github.com/rvlenth/emmeans" class="uri">https://github.com/rvlenth/emmeans</a>&gt;.</p>
<p>[8] R. Lenth. <em>lsmeans: Least-Squares Means</em>. R package version 2.30-0.
2018. &lt;URL: <a href="https://CRAN.R-project.org/package=lsmeans" class="uri">https://CRAN.R-project.org/package=lsmeans</a>&gt;.</p>
<p>[9] R. V. Lenth. “Least-Squares Means: The R Package lsmeans”. In:
<em>Journal of Statistical Software</em> 69.1 (2016), pp. 1-33. DOI:
10.18637/jss.v069.i01.</p>
<p>[10] E. Neuwirth. <em>RColorBrewer: ColorBrewer Palettes</em>. R package
version 1.1-2. 2014. &lt;URL:
<a href="https://CRAN.R-project.org/package=RColorBrewer" class="uri">https://CRAN.R-project.org/package=RColorBrewer</a>&gt;.</p>
<p>[11] E. Paradis, S. Blomberg, B. Bolker, et al. <em>ape: Analyses of
Phylogenetics and Evolution</em>. R package version 5.4-1. 2020. &lt;URL:
<a href="http://ape-package.ird.fr/" class="uri">http://ape-package.ird.fr/</a>&gt;.</p>
<p>[12] E. Paradis and K. Schliep. “ape 5.0: an environment for modern
phylogenetics and evolutionary analyses in R”. In: <em>Bioinformatics</em> 35
(2019), pp. 526-528.</p>
<p>[13] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[14] B. Ripley. <em>MASS: Support Functions and Datasets for Venables and
Ripley’s MASS</em>. R package version 7.3-53. 2020. &lt;URL:
<a href="http://www.stats.ox.ac.uk/pub/MASS4/" class="uri">http://www.stats.ox.ac.uk/pub/MASS4/</a>&gt;.</p>
<p>[15] M. R. Smith. <em>TreeTools: Create, Modify and Analyse Phylogenetic
Trees</em>. R package version 1.4.0. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=TreeTools" class="uri">https://CRAN.R-project.org/package=TreeTools</a>&gt;.</p>
<p>[16] W. N. Venables and B. D. Ripley. <em>Modern Applied Statistics with
S</em>. Fourth. ISBN 0-387-95457-0. New York: Springer, 2002. &lt;URL:
<a href="http://www.stats.ox.ac.uk/pub/MASS4/" class="uri">http://www.stats.ox.ac.uk/pub/MASS4/</a>&gt;.</p>
<p>[17] G. R. Warnes, B. Bolker, L. Bonebakker, et al. <em>gplots: Various R
Programming Tools for Plotting Data</em>. R package version 3.1.0. 2020.
&lt;URL: <a href="https://github.com/talgalili/gplots" class="uri">https://github.com/talgalili/gplots</a>&gt;.</p>
<p>[18] H. Wickham. <em>forcats: Tools for Working with Categorical Variables
(Factors)</em>. R package version 0.5.0. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=forcats" class="uri">https://CRAN.R-project.org/package=forcats</a>&gt;.</p>
<p>[19] H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>.
Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. &lt;URL:
<a href="https://ggplot2.tidyverse.org" class="uri">https://ggplot2.tidyverse.org</a>&gt;.</p>
<p>[20] H. Wickham. <em>stringr: Simple, Consistent Wrappers for Common
String Operations</em>. R package version 1.4.0. 2019. &lt;URL:
<a href="https://CRAN.R-project.org/package=stringr" class="uri">https://CRAN.R-project.org/package=stringr</a>&gt;.</p>
<p>[21] H. Wickham and J. Bryan. <em>usethis: Automate Package and Project
Setup</em>. R package version 1.6.3. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>&gt;.</p>
<p>[22] H. Wickham, W. Chang, L. Henry, et al. <em>ggplot2: Create Elegant
Data Visualisations Using the Grammar of Graphics</em>. R package version
3.3.2. 2020. &lt;URL: <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>&gt;.</p>
<p>[23] H. Wickham, J. Hester, and W. Chang. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.3.2. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>&gt;.</p>
<p>[24] C. O. Wilke. <em>ggridges: Ridgeline Plots in ggplot2</em>. R package
version 0.5.2. 2020. &lt;URL: <a href="https://wilkelab.org/ggridges" class="uri">https://wilkelab.org/ggridges</a>&gt;.</p>
<p>[25] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. &lt;URL: <a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[26] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.21. 2020. &lt;URL:
<a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[27] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[28] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.7. 2019. &lt;URL: <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>&gt;.</p>
<p>[29] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. &lt;URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[30] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.30. 2020. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[31] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>&gt;.</p>
<p>[32] Y. Xie, J. Cheng, and X. Tan. <em>DT: A Wrapper of the JavaScript
Library DataTables</em>. R package version 0.16. 2020. &lt;URL:
<a href="https://github.com/rstudio/DT" class="uri">https://github.com/rstudio/DT</a>&gt;.</p>
<p>[33] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020. &lt;URL:
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
<p>[34] G. Yu and T. T. Lam. <em>ggtree: an R package for visualization of
tree and annotation data</em>. R package version 2.0.4. 2020. &lt;URL:
<a href="https://yulab-smu.github.io/treedata-book/" class="uri">https://yulab-smu.github.io/treedata-book/</a>&gt;.</p>
<p>[35] G. Yu, T. T. Lam, H. Zhu, et al. “Two methods for mapping and
visualizing associated data on phylogeny using ggtree.” In: <em>Molecular
Biology and Evolution</em> 35 (2 2018), pp. 3041-3043. DOI:
10.1093/molbev/msy194. &lt;URL: <a href="https://doi.org/10.1093/molbev/msy194" class="uri">https://doi.org/10.1093/molbev/msy194</a>&gt;.</p>
<p>[36] G. Yu, D. Smith, H. Zhu, et al. “ggtree: an R package for
visualization and annotation of phylogenetic trees with their
covariates and other associated data.” In: <em>Methods in Ecology and
Evolution</em> 8 (1 2017), pp. 28-36. DOI: 10.1111/2041-210X.12628. &lt;URL:
<a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" class="uri">http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract</a>&gt;.</p>
<p>[37] A. Zeileis and G. Grothendieck. “zoo: S3 Infrastructure for
Regular and Irregular Time Series”. In: <em>Journal of Statistical
Software</em> 14.6 (2005), pp. 1-27. DOI: 10.18637/jss.v014.i06.</p>
<p>[38] A. Zeileis, G. Grothendieck, and J. A. Ryan. <em>zoo: S3
Infrastructure for Regular and Irregular Time Series (Z’s Ordered
Observations)</em>. R package version 1.8-8. 2020. &lt;URL:
<a href="http://zoo.R-Forge.R-project.org/" class="uri">http://zoo.R-Forge.R-project.org/</a>&gt;.</p>
<p>[39] A. Zeileis and T. Hothorn. “Diagnostic Checking in Regression
Relationships”. In: <em>R News</em> 2.3 (2002), pp. 7-10. &lt;URL:
<a href="https://CRAN.R-project.org/doc/Rnews/" class="uri">https://CRAN.R-project.org/doc/Rnews/</a>&gt;.</p>
<p>[40] A. Zeileis, C. Kleiber, and S. Jackman. “Regression Models for
Count Data in R”. In: <em>Journal of Statistical Software</em> 27.8 (2008).
&lt;URL: <a href="http://www.jstatsoft.org/v27/i08/" class="uri">http://www.jstatsoft.org/v27/i08/</a>&gt;.</p>
</div>
<div id="appendix-2" class="section level1 appendix">
<h1><span class="header-section-number">9</span> Appendix 2</h1>
<p>Version information about R, the operating system (OS) and attached or R loaded packages. This appendix was generated using <code>sessionInfo()</code>.</p>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] moments_0.14             forcats_0.5.0            TreeTools_1.4.0         
##  [4] ggridges_0.5.2           stringr_1.4.0            ape_5.4-1               
##  [7] ggtree_2.0.4             ggpubr_0.4.0             ggplot2_3.3.2           
## [10] chisq.posthoc.test_0.1.2 DT_0.16                  lsmeans_2.30-0          
## [13] emmeans_1.5.2-1          lmtest_0.9-38            zoo_1.8-8               
## [16] pscl_1.5.5               RColorBrewer_1.1-2       gplots_3.1.0            
## [19] devtools_2.3.2           usethis_1.6.3            formatR_1.7             
## [22] knitcitations_1.0.10     bookdown_0.21            rmarkdown_2.4           
## [25] knitr_1.30              
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.4-1    ggsignif_0.6.0      ellipsis_0.3.1     
##   [4] rio_0.5.16          rprojroot_1.3-2     estimability_1.3   
##   [7] fs_1.5.0            farver_2.0.3        remotes_2.2.0      
##  [10] bit64_4.0.5         fansi_0.4.1         mvtnorm_1.1-1      
##  [13] lubridate_1.7.9     RefManageR_1.2.12   xml2_1.3.2         
##  [16] codetools_0.2-16    R.methodsS3_1.8.1   pkgload_1.1.0      
##  [19] jsonlite_1.7.1      broom_0.7.1         R.oo_1.24.0        
##  [22] BiocManager_1.30.10 compiler_3.6.1      httr_1.4.2         
##  [25] rvcheck_0.1.8       backports_1.1.10    Matrix_1.2-18      
##  [28] assertthat_0.2.1    lazyeval_0.2.2      cli_2.1.0          
##  [31] htmltools_0.5.0     prettyunits_1.1.1   tools_3.6.1        
##  [34] igraph_1.2.6        coda_0.19-4         gtable_0.3.0       
##  [37] glue_1.4.2          dplyr_1.0.2         fastmatch_1.1-0    
##  [40] Rcpp_1.0.5          carData_3.0-4       cellranger_1.1.0   
##  [43] vctrs_0.3.4         nlme_3.1-149        gbRd_0.4-11        
##  [46] xfun_0.18           rbibutils_1.4       ps_1.4.0           
##  [49] openxlsx_4.2.2      testthat_2.3.2      lifecycle_0.2.0    
##  [52] phangorn_2.5.5      gtools_3.8.2        rstatix_0.6.0      
##  [55] MASS_7.3-53         scales_1.1.1        hms_0.5.3          
##  [58] parallel_3.6.1      yaml_2.2.1          curl_4.3           
##  [61] memoise_1.1.0       stringi_1.5.3       highr_0.8          
##  [64] desc_1.2.0          tidytree_0.3.3      caTools_1.18.0     
##  [67] pkgbuild_1.1.0      zip_2.1.1           bibtex_0.4.2.3     
##  [70] Rdpack_2.1          rlang_0.4.8         pkgconfig_2.0.3    
##  [73] bitops_1.0-6        evaluate_0.14       lattice_0.20-41    
##  [76] purrr_0.3.4         labeling_0.3        treeio_1.10.0      
##  [79] htmlwidgets_1.5.2   cowplot_1.1.0       bit_4.0.4          
##  [82] processx_3.4.4      tidyselect_1.1.0    plyr_1.8.6         
##  [85] magrittr_1.5        R6_2.4.1            generics_0.0.2     
##  [88] pillar_1.4.6        haven_2.3.1         foreign_0.8-75     
##  [91] withr_2.3.0         abind_1.4-5         tibble_3.0.4       
##  [94] crayon_1.3.4        car_3.0-10          KernSmooth_2.23-17 
##  [97] grid_3.6.1          readxl_1.3.1        data.table_1.12.8  
## [100] callr_3.5.1         digest_0.6.25       R.cache_0.14.0     
## [103] xtable_1.8-4        tidyr_1.1.2         R.utils_2.10.1     
## [106] munsell_0.5.0       quadprog_1.5-8      sessioninfo_1.1.1</code></pre>
</div>
<div id="appendix-3" class="section level1 appendix">
<h1><span class="header-section-number">10</span> Appendix 3</h1>
<p>Tissue culture data at the basis of analyses presented in step 3.</p>
<div id="htmlwidget-c0b7b66a143c6737679c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c0b7b66a143c6737679c">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574","575","576","577","578","579","580","581","582","583","584","585","586","587","588","589","590","591","592","593","594","595","596","597","598","599","600","601","602","603","604","605","606","607","608","609","610","611","612","613","614","615","616","617","618","619","620","621","622","623","624","625","626","627","628","629","630","631","632","633","634","635","636","637","638","639","640","641","642","643","644","645","646","647","648","649","650","651","652","653","654","655","656","657","658","659","660","661","662","663","664","665","666","667","668","669","670","671","672","673","674","675"],["1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5"],["IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control","IBA_05","IBA_05","IBA_05","IBA_1","IBA_1","IBA_1","NAA_05","NAA_05","NAA_05","NAA_1","NAA_1","NAA_1","Control","Control","Control"],["1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3","1","2","3"],["G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G2","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1","G1"],["b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b27_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b8_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_1","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b21_2","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b4_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b29_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b6_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","b26_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","18_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b15_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b3_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b13_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b12_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b10_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b7_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b17_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b28_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b19_1","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","b27_2","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","22_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b16_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","b11_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","29_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b24_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b9_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_1","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b1_2","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b5_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b25_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b20_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b2_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1","b18_1"],[0,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,0,0,0,1,1,1,1,1,1,1,0,1,0,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,1,1,1,1,1,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,0,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,0,1,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,0,1,1,1,1,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,1,1,1,0,1,1,1,1,1,1,1,0,0,0,0,1,0,1,0,1,0,1,0,1,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,1,0,0,0,0,0,0,1,1,1,0,0,0,0,1,1,1,1,1,1,1,1,1,1,0,1,0,0,0,1,1,0,1,1,1,1,1,1,1,1,1,1,0,0,0,1,0,0,0,1,0,0,0,0,1,0,0,0,0,1,0,0,1,1,1,1,1,1,1,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,1,1,1,1,1,0,0,0,0,1,1,1,1,0,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,0,0,1,1,1,1,0,0,1,1,1,0,0,0,1,1,1,1,1,1,1,1,0,1,1,1,0,0,0,1,0,1,1,1,1,0,1,0,1,0,1,0,0,0,0,1,1,1,1,1,1,0,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,1,0,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,1,0,1,1,0,1,1,1,1,1,0,0,0,1,0,1,1,1,0,1,0,1,1,1,1,0,0,0,1,1,1,1,0,1,0,1,0,1,1,1,0,1,0,0,0,0,0,0,0,1,1,1,1,0,0,0,0,0,1,1,1,1,1,1,1,1,0,1,1,1,0,0,0,1,0,1,1,1,1,1,1,1,1,1,1,0,0,0,1,1,0,1,0,1,1,1,0,1,1,0,0,0,0,1,0,1,1,1,1,1,1,1,1,1,1,0,0,0],[0,3,0,2,4,1,0,2,1,1,6,0,0,0,0,2,5,0,5,4,0,3,0,2,0,2,0,0,0,0,0,1,1,3,2,0,1,0,4,3,5,6,0,1,0,0,1,6,12,10,5,4,5,5,8,0,7,3,1,5,1,2,1,3,4,1,0,1,0,1,0,0,0,0,0,0,0,2,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,3,0,2,2,1,0,0,1,0,0,1,0,0,0,0,0,0,1,0,3,0,0,0,1,0,0,0,0,0,4,4,3,1,1,1,1,3,1,0,1,1,0,0,0,10,5,0,0,2,1,3,3,1,5,0,9,0,0,1,1,3,1,0,2,0,2,0,1,0,0,1,0,0,0,2,4,6,0,3,0,1,1,4,4,3,1,0,0,0,0,0,0,1,1,0,0,1,3,0,4,5,0,0,0,2,2,1,2,1,0,1,3,1,0,2,3,0,0,1,0,0,3,0,0,1,1,0,0,0,1,1,0,0,0,0,3,1,1,5,1,1,1,0,0,2,0,0,0,3,0,0,0,0,0,0,1,0,2,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,1,0,1,3,0,1,0,0,0,1,0,0,0,0,5,0,1,0,1,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,2,6,2,1,6,9,9,1,1,0,0,3,0,1,3,7,5,2,9,2,4,5,1,1,3,4,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,4,3,3,1,3,9,2,2,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,2,1,1,0,2,0,0,0,0,0,0,0,7,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,3,0,0,7,8,2,0,0,0,2,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,1,3,0,2,0,0,0,1,0,0,0,0,2,3,4,3,4,0,0,0,1,5,1,0,0,0,0,1,0,2,2,3,3,0,0,0,3,0,0,0,0,3,6,3,6,11,6,0,1,0,0,2,4,0,1,1,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,6,1,1,0,5,1,0,0,0,0,1,0,0,0,0,4,0,0,0,4,1,0,0,0,0,0,0,0,0,0,1,1,3,1,3,1,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,3,0,6,1,2,0,0,1,6,0,1,0,0,0,5,0,1,0,2,2,0,0,0,1,1,0,0,0,0,5,3,1,1,2,8,1,0,0,4,1,0,0,0,0,1,0,0,1,0,0,0,0,0,3,0,3,0,0,0],["G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G2_b20_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b19_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G1_b16_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b27_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b8_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b15_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G2_b26_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G1_b6_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G2_b2_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G1_b8_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G2_b4_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G1_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_1","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b21_2","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G2_b7_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b4_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G1_b29_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G2_b6_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G1_b26_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G2_18_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b5_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b15_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b3_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b13_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b12_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b10_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G1_b7_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b17_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b28_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b19_1","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G2_b27_2","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G1_22_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b16_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_b11_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_29_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b24_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G2_b25_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b9_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_1","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G1_b1_2","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G2_b5_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b25_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b20_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b2_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1","G1_b18_1"],[0,1,0,1,1,1,0,1,1,1,1,0,0,0,0,1,1,0,1,1,0,1,0,1,0,1,0,0,0,0,0,1,1,1,1,0,1,0,1,1,1,1,0,1,0,0,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,0,1,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,1,0,1,1,1,0,0,1,0,0,1,0,0,0,0,0,0,1,0,1,0,0,0,1,0,0,0,0,0,1,1,1,1,1,1,1,1,1,0,1,1,0,0,0,1,1,0,0,1,1,1,1,1,1,0,1,0,0,1,1,1,1,0,1,0,1,0,1,0,0,1,0,0,0,1,1,1,0,1,0,1,1,1,1,1,1,0,0,0,0,0,0,1,1,0,0,1,1,0,1,1,0,0,0,1,1,1,1,1,0,1,1,1,0,1,1,0,0,1,0,0,1,0,0,1,1,0,0,0,1,1,0,0,0,0,1,1,1,1,1,1,1,0,0,1,0,0,0,1,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,1,0,1,1,0,1,0,0,0,1,0,0,0,0,1,0,1,0,1,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,1,1,1,1,1,1,1,1,0,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,0,1,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,1,1,1,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,1,0,1,0,0,0,1,0,0,0,0,1,1,1,1,1,0,0,0,1,1,1,0,0,0,0,1,0,1,1,1,1,0,0,0,1,0,0,0,0,1,1,1,1,1,1,0,1,0,0,1,1,0,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,1,0,0,0,0,1,0,0,0,0,1,0,0,0,1,1,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,1,0,1,1,1,0,0,1,1,0,1,0,0,0,1,0,1,0,1,1,0,0,0,1,1,0,0,0,0,1,1,1,1,1,1,1,0,0,1,1,0,0,0,0,1,0,0,1,0,0,0,0,0,1,0,1,0,0,0],["pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","pink","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Block<\/th>\n      <th>Treatment<\/th>\n      <th>Replicate<\/th>\n      <th>Genotype<\/th>\n      <th>Individual<\/th>\n      <th>Callus<\/th>\n      <th>Root<\/th>\n      <th>IndGeno<\/th>\n      <th>RootBin<\/th>\n      <th>ClusterRoot<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"columnDefs":[{"className":"dt-right","targets":[6,7,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="appendix-4" class="section level1 appendix">
<h1><span class="header-section-number">11</span> Appendix 4</h1>
<p><em>In vitro</em> survival and growth data for plantlets at the basis of analyses presented in step 4.</p>
<div id="htmlwidget-8d8b32e925bf0fa476eb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8d8b32e925bf0fa476eb">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273"],["G2_b20_1","G1_b19_1","G1_b16_1","G2_b27_1","G2_b8_1","G2_b15_1","G2_b26_1","G2_b2_1","G2_b20_1","G1_b19_1","G1_b16_1","G2_b27_1","G2_b15_1","G2_b20_1","G1_b19_1","G1_b16_1","G2_b27_1","G2_b8_1","G1_b6_1","G1_b16_1","G2_b27_1","G2_b20_1","G1_b16_1","G2_b27_1","G2_b8_1","G2_b2_1","G2_b20_1","G2_b27_1","G2_b8_1","G2_b27_1","G2_b27_1","G1_b19_1","G2_b8_1","G2_b20_1","G2_b27_1","G1_b6_1","G2_b2_1","G1_b16_1","G2_b27_1","G2_b8_1","G2_b15_1","G1_b6_1","G2_b20_1","G1_b19_1","G1_b16_1","G2_b27_1","G2_b8_1","G1_b6_1","G2_b2_1","G1_b16_1","G2_b27_1","G2_b8_1","G2_b15_1","G1_b8_1","G1_b21_1","G2_b21_1","G2_b7_1","G1_b4_1","G1_b29_1","G1_b8_1","G2_b4_1","G1_b21_1","G2_b21_1","G2_b7_1","G1_b4_1","G1_b29_1","G2_b6_1","G1_b8_1","G2_b21_2","G2_b7_1","G1_b4_1","G1_b29_1","G1_b8_1","G2_b4_1","G1_b21_1","G2_b21_1","G2_b21_2","G2_b7_1","G1_b29_1","G1_b8_1","G2_b4_1","G1_b21_1","G2_b21_1","G2_b21_2","G2_b7_1","G1_b29_1","G1_b8_1","G2_b4_1","G2_b21_1","G2_b21_2","G2_b7_1","G1_b29_1","G2_b4_1","G2_b7_1","G1_b29_1","G1_b8_1","G2_b4_1","G1_b21_1","G2_b21_1","G2_b21_2","G2_b7_1","G2_b6_1","G2_b4_1","G2_b21_1","G2_b7_1","G1_b8_1","G2_b4_1","G1_b21_1","G2_b21_1","G2_b7_1","G1_b8_1","G2_b4_1","G1_b4_1","G1_b29_1","G1_b8_1","G2_b21_1","G2_b21_2","G2_b7_1","G1_b4_1","G1_b29_1","G1_b8_1","G2_b4_1","G1_b21_1","G2_b21_1","G2_b21_2","G2_b7_1","G1_b4_1","G2_b6_1","G1_b15_1","G1_b13_1","G1_b10_1","G1_b3_1","G1_b13_1","G1_b26_1","G1_b5_1","G1_b3_1","G1_b13_1","G1_b10_1","G1_b26_1","G2_18_1","G1_b5_1","G1_b3_1","G1_b13_1","G1_b10_1","G1_b3_1","G1_b13_1","G1_b3_1","G2_18_1","G1_b3_1","G1_b13_1","G1_b10_1","G1_b5_1","G1_b3_1","G1_b13_1","G1_b10_1","G1_b5_1","G1_b3_1","G1_b13_1","G1_b10_1","G2_18_1","G1_b3_1","G1_b13_1","G1_b10_1","G2_18_1","G1_b5_1","G1_b3_1","G1_b26_1","G2_18_1","G1_b5_1","G1_b3_1","G1_b13_1","G1_b10_1","G1_b5_1","G1_b3_1","G1_b13_1","G1_b10_1","G1_b13_1","G2_b28_1","G2_b27_2","G2_b11_1","G2_29_1","G2_b24_1","G2_b17_1","G2_b28_1","G2_b27_2","G2_b16_1","G2_b11_1","G2_b24_1","G2_b11_1","G2_29_1","G2_b24_1","G2_b28_1","G2_b27_2","G2_b16_1","G2_b11_1","G2_29_1","G2_b24_1","G2_b17_1","G2_b11_1","G2_29_1","G2_b24_1","G2_b17_1","G2_b16_1","G2_b16_1","G2_b11_1","G2_b24_1","G2_b17_1","G2_b27_2","G2_b16_1","G2_b11_1","G2_29_1","G2_b24_1","G2_b24_1","G2_b17_1","G2_b16_1","G2_b11_1","G2_29_1","G2_b24_1","G2_b17_1","G2_b19_1","G2_b27_2","G2_b16_1","G2_b11_1","G2_b24_1","G2_b27_2","G2_b24_1","G1_b9_1","G1_b20_1","G1_b2_1","G1_b1_1","G1_b9_1","G1_b1_1","G1_b1_2","G1_b25_1","G1_b20_1","G1_b2_1","G1_b25_1","G1_b20_1","G1_b2_1","G1_b18_1","G1_b9_1","G1_b1_1","G1_b1_2","G1_b25_1","G1_b20_1","G1_b2_1","G2_b5_1","G1_b2_1","G1_b9_1","G1_b1_2","G1_b20_1","G1_b2_1","G1_b25_1","G1_b18_1","G1_b9_1","G1_b1_2","G1_b25_1","G1_b2_1","G1_b1_2","G2_b5_1","G1_b25_1","G1_b1_2","G1_b25_1","G1_b2_1","G1_b18_1","G2_b25_1","G1_b9_1","G1_b1_1","G1_b1_2","G1_b25_1","G1_b20_1","G1_b2_1","G1_b18_1"],["pink","pink","blue","blue","pink","black","black","black","pink","pink","blue","blue","black","pink","pink","blue","blue","pink",null,"blue","blue","pink","blue","blue","pink",null,"pink","blue","pink","blue","blue","pink","pink","pink","blue",null,null,"blue","blue","pink","black",null,"pink","pink","blue","blue","pink",null,null,"blue","blue","pink","black","blue","pink","blue","blue","pink","blue","blue","blue","pink","blue","blue","pink","blue","black","blue","pink","blue","pink","blue","blue","blue","pink","blue","pink","blue","blue","blue","blue","pink","blue","pink","blue","blue","blue","blue","blue","pink","blue","blue","blue","blue","blue","blue","blue","pink","blue","pink","blue","black","blue","blue","blue","blue","blue","pink","blue","blue","blue","blue","pink","blue","blue","blue","pink","blue","pink","blue","blue","blue","pink","blue","pink","blue","pink","black","black","blue","blue","blue","blue","black","pink","blue","blue","blue","black","pink","pink","blue","blue","blue","blue","blue","blue","pink","blue","blue","blue","pink","blue","blue","blue","pink","blue","blue","blue","pink","blue","blue","blue","pink","pink","blue","black","pink","pink","blue","blue","blue","pink","blue","blue","blue","blue","black","pink","blue","pink","blue","pink","black","pink","pink","blue","blue","blue","pink","blue","black","pink","pink","blue","pink","blue","pink","blue","pink","blue","pink","pink","pink","blue","blue","pink","pink","pink","blue","pink","blue","blue","pink","pink","blue","pink","blue","pink","black","pink","pink","blue","blue","pink","blue","pink","pink","blue","black","pink","black","pink","pink","pink","blue","pink","pink","blue","black","pink","black","pink","pink","pink","blue","black","blue","pink","pink","pink","blue","pink","black","pink","pink","pink","blue","pink","black","pink","pink","pink","blue","black","black","pink","black","pink","pink","pink","blue","black"],[1,1,1,0,0,0,0,0.5,0.5,1,1,0.5,1,1,1,0.5,0.5,0,0,1,0,1,1,1,0,1,0.5,1,0,1,0,0,0,0.5,0.5,0,1,1,1,0,0.5,0,0.5,1,0,0.5,0.5,0,0,0,1,0,0.5,1,0,0.5,1,1,1,0,0.5,0,0,0.5,0.5,0,1,0,1,0.5,0.5,1,0,1,0,1,0,0.5,1,1,1,1,1,0,1,1,0,0.5,0,0,1,0.5,0,0,1,0.5,0.5,0.5,1,1,0.5,1,1,0,0.5,1,1,0,0,0.5,0,0,0.5,0.5,0.5,1,0,0.5,0,0.5,0,1,0,0,1,1,1,1,0,1,1,1,1,0.5,1,1,1,1,1,1,0,1,1,1,1,0.5,1,0,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,0,0,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.5,1,1,0.5,1,1,0,1,0.5,1,1,1,0.5,1,0.5,0.5,1,1,0,0,1,1,0,0,1,1,1,0.5,1,0,1,0,1,0.5,1,1,1,1,1,1,1,1,1,1,1,1,0,1,0,1,0.5,1,1,1,1,1,1,1,1,1,1,1,1,0,1,0.5,1,0,1,0.5,1,1,1,0.5,0.5,1,0.5,1,0.5,1,1,0,0,1,1,0,1,0.5,1,1],[3.5,2.3,2.8,0,0,0,0,1.9,1.4,1.7,2.4,2,2.3,2.7,2.4,2,2.4,0,0,2.7,0,2.7,2.7,4.4,0,3.5,1.5,3.7,0,3.7,0,0,0,1,1.5,0,3.7,3,3.8,0,1.5,0,2.4,2.5,0,2.5,1.1,0,0,0,3.9,0,1.5,2.7,0,1.5,2.5,2.3,3.4,0,1.5,0,0,1.9,1.5,0,2,0,1.9,1.5,2,2.5,0,2.5,0,2,0,1,3,2,4.1,2.8,1.9,0,2.8,3.6,0,1.2,0,0,3.5,1.9,0,0,3.6,1.3,1.7,1.1,2,1.9,1.4,2.6,3,0,2.2,2.4,3.4,0,0,2,0,0,2,1.4,1.2,2.1,0,1.5,0,1.4,0,3.6,0,0,2.1,3,2,2.2,0,2.4,3.1,2,2.7,1.5,3.3,3.5,3,2.9,1.7,2.6,0,3.6,2.5,3.1,1.7,1.6,2.4,0,3.5,2.4,3.2,2.5,2.9,3,2.5,2.9,2,2.5,1.5,0,3.1,3.1,3,0,0,3.5,2.4,0,3.1,2,2.5,3.4,2.2,3,2.7,2.5,2.6,3.2,2.5,2.6,2.4,2.3,2.3,1.1,2.1,1.7,1,2.4,2.8,0,4,1,2.8,2.9,1.9,1.2,3.1,1.1,1,3.6,3.9,0,0,2.3,1.2,0,0,1.7,3.3,1.9,2,2.4,0,2.7,0,1.4,1.5,3,2,3.5,2.1,1.5,1.8,3.2,2,1.5,2.4,1.9,2.9,0,2.5,0,2.6,1.1,2.5,3.6,2.9,1.9,3.1,2.8,2.5,1.5,3.2,2,1.8,2,0,2.4,1.1,2.7,0,2,1,2.5,1.1,2.5,1,1,2.6,1.1,2.2,1,3,1.5,0,0,2.1,2.7,0,2.5,1.2,2.8,2.7],[1,1,1,0,0,0,0,0,0,1,1,0,1,1,1,0,0,0,0,1,0,1,1,1,0,1,0,1,0,1,0,0,0,0,0,0,1,1,1,0,0,0,0.5,1,0,0,0,0,0,0,1,0,0,1,0,0,1,1,0,0,0,0,0,0,0.5,0,1,0,0.5,0.5,0,1,0,1,0,0.5,0,0,1,1,1,1,0,0,1,1,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,1,1,0,0,1,1,0,0,0,0,0,0,0,0,0.5,0,0,0,0,0,1,0,0,1,1,0.5,1,0,1,1,1,1,0,1,1,1,1,0,1,0,1,1,1,0,0,1,0,1,1,1,1,1,1,1,1,1,1,1,0,1,1,0,0,0,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,0,1,1,0,0.5,0,1,0.5,0,0,1,0,0,1,1,0,0,0,0,0,0,1,1,1,0,1,0,1,0,1,0,1,1,1,1,0,1,0.5,0.5,0,1,1,1,0,1,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1,0,1,0,1,0,1,0,1,0,1,0,0,1,0,1,0,1,1,0,0,1,1,0,1,0,1,1],[4.5,2.7,2.8,0,0,0,0,0,0,2.5,2.7,0,3,3.5,2.8,0,0,0,0,3,0,3.5,3.2,4.6,0,4,0,4.5,0,4.5,0,0,0,0,0,0,4.4,3,4,0,0,0,2.9,3,0,0,0,0,0,0,4,0,0,3.5,0,0,3.7,3,0,0,0,0,0,0,2,0,3.5,0,2,2,0,3.4,0,4,0,2,0,0,3,2.8,4.5,2.8,0,0,3,3.6,0,0,0,0,4,0,0,0,3.1,0,0,0,0,0,0,4.8,4,0,0,3.4,4,0,0,0,0,0,0,0,0,2.7,0,0,0,0,0,3.7,0,0,2.7,4,2.7,3.1,0,3,3.5,3,3,0,3.3,3.2,3.2,3.1,0,3,0,3,2.9,3,0,0,3,0,3.8,3,2.9,3,3.4,3,3.1,3.1,2.9,2.5,3.6,0,3.6,3.5,0,0,0,3.5,2.3,0,3.3,3.2,2.6,3.2,3,3.5,3.4,3.7,3.7,3.9,2.7,3.3,3.6,3.6,3.4,0,3,1.8,0,2.5,3.4,0,3,0,2.9,3.9,0,0,3.6,0,0,4,4,0,0,0,0,0,0,2.3,4.7,2.6,0,3.2,0,3.5,0,3.1,0,3.5,3.5,3.5,3,0,2,3,1.9,0,2.6,1.5,3.2,0,2.5,0,2.7,0,4,3.4,2.5,3,3.4,3.5,3.3,1.5,3.3,2.6,2.6,3,0,3,0,2.9,0,3.5,0,3.1,0,2.9,0,0,3,0,3.2,0,2.2,1.5,0,0,2.5,2.9,0,2.6,0,2.9,3.1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SeedID<\/th>\n      <th>Cluster<\/th>\n      <th>X3_w_survival<\/th>\n      <th>X3_w_height<\/th>\n      <th>X5_w_survival<\/th>\n      <th>X5_w_height<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"columnDefs":[{"className":"dt-right","targets":[3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
